// @kinde-oss/kinde-typescript-sdk/dist/models/AddAPIsRequest.js
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
/**
 * Check if a given object implements the AddAPIsRequest interface.
 */
export function instanceOfAddAPIsRequest(value) {
    var isInstance = true;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "audience" in value;
    return isInstance;
}
export function AddAPIsRequestFromJSON(json) {
    return AddAPIsRequestFromJSONTyped(json, false);
}
export function AddAPIsRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': json['name'],
        'audience': json['audience'],
    };
}
export function AddAPIsRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'audience': value.audience,
    };
}
