// @kinde-oss/kinde-typescript-sdk/dist/models/AddOrganizationUsersRequest.js
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime.js';
import { AddOrganizationUsersRequestUsersInnerFromJSON, AddOrganizationUsersRequestUsersInnerToJSON, } from './AddOrganizationUsersRequestUsersInner.js';
/**
 * Check if a given object implements the AddOrganizationUsersRequest interface.
 */
export function instanceOfAddOrganizationUsersRequest(value) {
    var isInstance = true;
    return isInstance;
}
export function AddOrganizationUsersRequestFromJSON(json) {
    return AddOrganizationUsersRequestFromJSONTyped(json, false);
}
export function AddOrganizationUsersRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'users': !exists(json, 'users') ? undefined : (json['users'].map(AddOrganizationUsersRequestUsersInnerFromJSON)),
    };
}
export function AddOrganizationUsersRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'users': value.users === undefined ? undefined : (value.users.map(AddOrganizationUsersRequestUsersInnerToJSON)),
    };
}
