// @kinde-oss/kinde-typescript-sdk/dist/models/Api.js
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime.js';
import { ApiApplicationsInnerFromJSON, ApiApplicationsInnerToJSON, } from './ApiApplicationsInner.js';
/**
 * Check if a given object implements the Api interface.
 */
export function instanceOfApi(value) {
    var isInstance = true;
    return isInstance;
}
export function ApiFromJSON(json) {
    return ApiFromJSONTyped(json, false);
}
export function ApiFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'id': !exists(json, 'id') ? undefined : json['id'],
        'code': !exists(json, 'code') ? undefined : json['code'],
        'name': !exists(json, 'name') ? undefined : json['name'],
        'message': !exists(json, 'message') ? undefined : json['message'],
        'audience': !exists(json, 'audience') ? undefined : json['audience'],
        'applications': !exists(json, 'applications') ? undefined : (json['applications'].map(ApiApplicationsInnerFromJSON)),
    };
}
export function ApiToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'id': value.id,
        'code': value.code,
        'name': value.name,
        'message': value.message,
        'audience': value.audience,
        'applications': value.applications === undefined ? undefined : (value.applications.map(ApiApplicationsInnerToJSON)),
    };
}
