// @kinde-oss/kinde-typescript-sdk/dist/models/Apis.js
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime.js';
/**
 * Check if a given object implements the Apis interface.
 */
export function instanceOfApis(value) {
    var isInstance = true;
    return isInstance;
}
export function ApisFromJSON(json) {
    return ApisFromJSONTyped(json, false);
}
export function ApisFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'id': !exists(json, 'id') ? undefined : json['id'],
        'name': !exists(json, 'name') ? undefined : json['name'],
        'audience': !exists(json, 'audience') ? undefined : json['audience'],
        'isManagementApi': !exists(json, 'is_management_api') ? undefined : json['is_management_api'],
    };
}
export function ApisToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'id': value.id,
        'name': value.name,
        'audience': value.audience,
        'is_management_api': value.isManagementApi,
    };
}
