// @kinde-oss/kinde-typescript-sdk/dist/models/ConnectedAppsAuthUrl.js
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime.js';
/**
 * Check if a given object implements the ConnectedAppsAuthUrl interface.
 */
export function instanceOfConnectedAppsAuthUrl(value) {
    var isInstance = true;
    return isInstance;
}
export function ConnectedAppsAuthUrlFromJSON(json) {
    return ConnectedAppsAuthUrlFromJSONTyped(json, false);
}
export function ConnectedAppsAuthUrlFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'url': !exists(json, 'url') ? undefined : json['url'],
        'sessionId': !exists(json, 'session_id') ? undefined : json['session_id'],
    };
}
export function ConnectedAppsAuthUrlToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'url': value.url,
        'session_id': value.sessionId,
    };
}
