// @kinde-oss/kinde-typescript-sdk/dist/models/ConnectedAppsAccessToken.js
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime.js';
/**
 * Check if a given object implements the ConnectedAppsAccessToken interface.
 */
export function instanceOfConnectedAppsAccessToken(value) {
    var isInstance = true;
    return isInstance;
}
export function ConnectedAppsAccessTokenFromJSON(json) {
    return ConnectedAppsAccessTokenFromJSONTyped(json, false);
}
export function ConnectedAppsAccessTokenFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'accessToken': !exists(json, 'access_token') ? undefined : json['access_token'],
        'accessTokenExpiry': !exists(json, 'access_token_expiry') ? undefined : json['access_token_expiry'],
    };
}
export function ConnectedAppsAccessTokenToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'access_token': value.accessToken,
        'access_token_expiry': value.accessTokenExpiry,
    };
}
