// @kinde-oss/kinde-typescript-sdk/dist/models/CreateFeatureFlagRequest.js
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime.js';
/**
 * @export
 */
export var CreateFeatureFlagRequestTypeEnum = {
    Str: 'str',
    Int: 'int',
    Bool: 'bool'
};
/**
 * @export
 */
export var CreateFeatureFlagRequestAllowOverrideLevelEnum = {
    Env: 'env',
    Org: 'org',
    Usr: 'usr'
};
/**
 * Check if a given object implements the CreateFeatureFlagRequest interface.
 */
export function instanceOfCreateFeatureFlagRequest(value) {
    var isInstance = true;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "key" in value;
    isInstance = isInstance && "type" in value;
    isInstance = isInstance && "defaultValue" in value;
    return isInstance;
}
export function CreateFeatureFlagRequestFromJSON(json) {
    return CreateFeatureFlagRequestFromJSONTyped(json, false);
}
export function CreateFeatureFlagRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': json['name'],
        'description': !exists(json, 'description') ? undefined : json['description'],
        'key': json['key'],
        'type': json['type'],
        'allowOverrideLevel': !exists(json, 'allow_override_level') ? undefined : json['allow_override_level'],
        'defaultValue': json['default_value'],
    };
}
export function CreateFeatureFlagRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'description': value.description,
        'key': value.key,
        'type': value.type,
        'allow_override_level': value.allowOverrideLevel,
        'default_value': value.defaultValue,
    };
}
