// @kinde-oss/kinde-typescript-sdk/dist/models/CreateUserRequestIdentitiesInner.js
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime.js';
import { CreateUserRequestIdentitiesInnerDetailsFromJSON, CreateUserRequestIdentitiesInnerDetailsToJSON, } from './CreateUserRequestIdentitiesInnerDetails.js';
/**
 * @export
 */
export var CreateUserRequestIdentitiesInnerTypeEnum = {
    Email: 'email'
};
/**
 * Check if a given object implements the CreateUserRequestIdentitiesInner interface.
 */
export function instanceOfCreateUserRequestIdentitiesInner(value) {
    var isInstance = true;
    return isInstance;
}
export function CreateUserRequestIdentitiesInnerFromJSON(json) {
    return CreateUserRequestIdentitiesInnerFromJSONTyped(json, false);
}
export function CreateUserRequestIdentitiesInnerFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'type': !exists(json, 'type') ? undefined : json['type'],
        'details': !exists(json, 'details') ? undefined : CreateUserRequestIdentitiesInnerDetailsFromJSON(json['details']),
    };
}
export function CreateUserRequestIdentitiesInnerToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'type': value.type,
        'details': CreateUserRequestIdentitiesInnerDetailsToJSON(value.details),
    };
}
