// @kinde-oss/kinde-typescript-sdk/dist/models/CreateUserResponse.js
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime.js';
import { UserIdentityFromJSON, UserIdentityToJSON, } from './UserIdentity.js';
/**
 * Check if a given object implements the CreateUserResponse interface.
 */
export function instanceOfCreateUserResponse(value) {
    var isInstance = true;
    return isInstance;
}
export function CreateUserResponseFromJSON(json) {
    return CreateUserResponseFromJSONTyped(json, false);
}
export function CreateUserResponseFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'id': !exists(json, 'id') ? undefined : json['id'],
        'created': !exists(json, 'created') ? undefined : json['created'],
        'identities': !exists(json, 'identities') ? undefined : (json['identities'].map(UserIdentityFromJSON)),
    };
}
export function CreateUserResponseToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'id': value.id,
        'created': value.created,
        'identities': value.identities === undefined ? undefined : (value.identities.map(UserIdentityToJSON)),
    };
}
