// @kinde-oss/kinde-typescript-sdk/dist/models/CreateOrganizationUserPermissionRequest.js
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime.js';
/**
 * Check if a given object implements the CreateOrganizationUserPermissionRequest interface.
 */
export function instanceOfCreateOrganizationUserPermissionRequest(value) {
    var isInstance = true;
    return isInstance;
}
export function CreateOrganizationUserPermissionRequestFromJSON(json) {
    return CreateOrganizationUserPermissionRequestFromJSONTyped(json, false);
}
export function CreateOrganizationUserPermissionRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'permissionId': !exists(json, 'permission_id') ? undefined : json['permission_id'],
    };
}
export function CreateOrganizationUserPermissionRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'permission_id': value.permissionId,
    };
}
