// @kinde-oss/kinde-typescript-sdk/dist/models/CreateOrganizationRequest.js
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime.js';
/**
 * @export
 */
export var CreateOrganizationRequestFeatureFlagsEnum = {
    Str: 'str',
    Int: 'int',
    Bool: 'bool'
};
/**
 * Check if a given object implements the CreateOrganizationRequest interface.
 */
export function instanceOfCreateOrganizationRequest(value) {
    var isInstance = true;
    return isInstance;
}
export function CreateOrganizationRequestFromJSON(json) {
    return CreateOrganizationRequestFromJSONTyped(json, false);
}
export function CreateOrganizationRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': !exists(json, 'name') ? undefined : json['name'],
        'featureFlags': !exists(json, 'feature_flags') ? undefined : json['feature_flags'],
        'externalId': !exists(json, 'external_id') ? undefined : json['external_id'],
        'backgroundColor': !exists(json, 'background_color') ? undefined : json['background_color'],
        'buttonColor': !exists(json, 'button_color') ? undefined : json['button_color'],
        'buttonTextColor': !exists(json, 'button_text_color') ? undefined : json['button_text_color'],
        'linkColor': !exists(json, 'link_color') ? undefined : json['link_color'],
        'handle': !exists(json, 'handle') ? undefined : json['handle'],
    };
}
export function CreateOrganizationRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'feature_flags': value.featureFlags,
        'external_id': value.externalId,
        'background_color': value.backgroundColor,
        'button_color': value.buttonColor,
        'button_text_color': value.buttonTextColor,
        'link_color': value.linkColor,
        'handle': value.handle,
    };
}
