// @kinde-oss/kinde-typescript-sdk/dist/models/CreateApplicationRequest.js
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime.js';
/**
 * @export
 */
export var CreateApplicationRequestTypeEnum = {
    Reg: 'reg',
    Spa: 'spa',
    M2m: 'm2m'
};
/**
 * Check if a given object implements the CreateApplicationRequest interface.
 */
export function instanceOfCreateApplicationRequest(value) {
    var isInstance = true;
    return isInstance;
}
export function CreateApplicationRequestFromJSON(json) {
    return CreateApplicationRequestFromJSONTyped(json, false);
}
export function CreateApplicationRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': !exists(json, 'name') ? undefined : json['name'],
        'type': !exists(json, 'type') ? undefined : json['type'],
    };
}
export function CreateApplicationRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'type': value.type,
    };
}
