// @kinde-oss/kinde-typescript-sdk/dist/models/CreateUserRequestIdentitiesInnerDetails.js
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime.js';
/**
 * Check if a given object implements the CreateUserRequestIdentitiesInnerDetails interface.
 */
export function instanceOfCreateUserRequestIdentitiesInnerDetails(value) {
    var isInstance = true;
    return isInstance;
}
export function CreateUserRequestIdentitiesInnerDetailsFromJSON(json) {
    return CreateUserRequestIdentitiesInnerDetailsFromJSONTyped(json, false);
}
export function CreateUserRequestIdentitiesInnerDetailsFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'email': !exists(json, 'email') ? undefined : json['email'],
    };
}
export function CreateUserRequestIdentitiesInnerDetailsToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'email': value.email,
    };
}
