// @kinde-oss/kinde-typescript-sdk/dist/models/CreateSubscriberSuccessResponse.js
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime.js';
import { CreateSubscriberSuccessResponseSubscriberFromJSON, CreateSubscriberSuccessResponseSubscriberToJSON, } from './CreateSubscriberSuccessResponseSubscriber.js';
/**
 * Check if a given object implements the CreateSubscriberSuccessResponse interface.
 */
export function instanceOfCreateSubscriberSuccessResponse(value) {
    var isInstance = true;
    return isInstance;
}
export function CreateSubscriberSuccessResponseFromJSON(json) {
    return CreateSubscriberSuccessResponseFromJSONTyped(json, false);
}
export function CreateSubscriberSuccessResponseFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'subscriber': !exists(json, 'subscriber') ? undefined : CreateSubscriberSuccessResponseSubscriberFromJSON(json['subscriber']),
    };
}
export function CreateSubscriberSuccessResponseToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'subscriber': CreateSubscriberSuccessResponseSubscriberToJSON(value.subscriber),
    };
}
