// @kinde-oss/kinde-typescript-sdk/dist/models/CreateRoleRequest.js
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime.js';
/**
 * Check if a given object implements the CreateRoleRequest interface.
 */
export function instanceOfCreateRoleRequest(value) {
    var isInstance = true;
    return isInstance;
}
export function CreateRoleRequestFromJSON(json) {
    return CreateRoleRequestFromJSONTyped(json, false);
}
export function CreateRoleRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': !exists(json, 'name') ? undefined : json['name'],
        'description': !exists(json, 'description') ? undefined : json['description'],
        'key': !exists(json, 'key') ? undefined : json['key'],
        'isDefaultRole': !exists(json, 'is_default_role') ? undefined : json['is_default_role'],
    };
}
export function CreateRoleRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'description': value.description,
        'key': value.key,
        'is_default_role': value.isDefaultRole,
    };
}
