// @kinde-oss/kinde-typescript-sdk/dist/models/CreateUserRequestProfile.js
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime.js';
/**
 * Check if a given object implements the CreateUserRequestProfile interface.
 */
export function instanceOfCreateUserRequestProfile(value) {
    var isInstance = true;
    return isInstance;
}
export function CreateUserRequestProfileFromJSON(json) {
    return CreateUserRequestProfileFromJSONTyped(json, false);
}
export function CreateUserRequestProfileFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'givenName': !exists(json, 'given_name') ? undefined : json['given_name'],
        'familyName': !exists(json, 'family_name') ? undefined : json['family_name'],
    };
}
export function CreateUserRequestProfileToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'given_name': value.givenName,
        'family_name': value.familyName,
    };
}
