// @kinde-oss/kinde-typescript-sdk/dist/models/CreateOrganizationResponse.js
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime.js';
import { CreateOrganizationResponseOrganizationFromJSON, CreateOrganizationResponseOrganizationToJSON, } from './CreateOrganizationResponseOrganization.js';
/**
 * Check if a given object implements the CreateOrganizationResponse interface.
 */
export function instanceOfCreateOrganizationResponse(value) {
    var isInstance = true;
    return isInstance;
}
export function CreateOrganizationResponseFromJSON(json) {
    return CreateOrganizationResponseFromJSONTyped(json, false);
}
export function CreateOrganizationResponseFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'message': !exists(json, 'message') ? undefined : json['message'],
        'code': !exists(json, 'code') ? undefined : json['code'],
        'organization': !exists(json, 'organization') ? undefined : CreateOrganizationResponseOrganizationFromJSON(json['organization']),
    };
}
export function CreateOrganizationResponseToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'message': value.message,
        'code': value.code,
        'organization': CreateOrganizationResponseOrganizationToJSON(value.organization),
    };
}
