// @kinde-oss/kinde-typescript-sdk/dist/models/CreateOrganizationUserRoleRequest.js
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime.js';
/**
 * Check if a given object implements the CreateOrganizationUserRoleRequest interface.
 */
export function instanceOfCreateOrganizationUserRoleRequest(value) {
    var isInstance = true;
    return isInstance;
}
export function CreateOrganizationUserRoleRequestFromJSON(json) {
    return CreateOrganizationUserRoleRequestFromJSONTyped(json, false);
}
export function CreateOrganizationUserRoleRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'roleId': !exists(json, 'role_id') ? undefined : json['role_id'],
    };
}
export function CreateOrganizationUserRoleRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'role_id': value.roleId,
    };
}
