// @kinde-oss/kinde-typescript-sdk/dist/models/CreateSubscriberSuccessResponseSubscriber.js
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime.js';
/**
 * Check if a given object implements the CreateSubscriberSuccessResponseSubscriber interface.
 */
export function instanceOfCreateSubscriberSuccessResponseSubscriber(value) {
    var isInstance = true;
    return isInstance;
}
export function CreateSubscriberSuccessResponseSubscriberFromJSON(json) {
    return CreateSubscriberSuccessResponseSubscriberFromJSONTyped(json, false);
}
export function CreateSubscriberSuccessResponseSubscriberFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'subscriberId': !exists(json, 'subscriber_id') ? undefined : json['subscriber_id'],
    };
}
export function CreateSubscriberSuccessResponseSubscriberToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'subscriber_id': value.subscriberId,
    };
}
