// @kinde-oss/kinde-typescript-sdk/dist/models/ErrorResponse.js
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime.js';
import { ModelErrorFromJSON, ModelErrorToJSON, } from './ModelError.js';
/**
 * Check if a given object implements the ErrorResponse interface.
 */
export function instanceOfErrorResponse(value) {
    var isInstance = true;
    return isInstance;
}
export function ErrorResponseFromJSON(json) {
    return ErrorResponseFromJSONTyped(json, false);
}
export function ErrorResponseFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'errors': !exists(json, 'errors') ? undefined : (json['errors'].map(ModelErrorFromJSON)),
    };
}
export function ErrorResponseToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'errors': value.errors === undefined ? undefined : (value.errors.map(ModelErrorToJSON)),
    };
}
