// @kinde-oss/kinde-typescript-sdk/dist/models/GetApplicationResponse.js
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime.js';
import { GetApplicationResponseApplicationFromJSON, GetApplicationResponseApplicationToJSON, } from './GetApplicationResponseApplication.js';
/**
 * Check if a given object implements the GetApplicationResponse interface.
 */
export function instanceOfGetApplicationResponse(value) {
    var isInstance = true;
    return isInstance;
}
export function GetApplicationResponseFromJSON(json) {
    return GetApplicationResponseFromJSONTyped(json, false);
}
export function GetApplicationResponseFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'code': !exists(json, 'code') ? undefined : json['code'],
        'message': !exists(json, 'message') ? undefined : json['message'],
        'application': !exists(json, 'application') ? undefined : GetApplicationResponseApplicationFromJSON(json['application']),
    };
}
export function GetApplicationResponseToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'code': value.code,
        'message': value.message,
        'application': GetApplicationResponseApplicationToJSON(value.application),
    };
}
