// @kinde-oss/kinde-typescript-sdk/dist/models/GetApplicationsResponse.js
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime.js';
import { ApplicationsFromJSON, ApplicationsToJSON, } from './Applications.js';
/**
 * Check if a given object implements the GetApplicationsResponse interface.
 */
export function instanceOfGetApplicationsResponse(value) {
    var isInstance = true;
    return isInstance;
}
export function GetApplicationsResponseFromJSON(json) {
    return GetApplicationsResponseFromJSONTyped(json, false);
}
export function GetApplicationsResponseFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'code': !exists(json, 'code') ? undefined : json['code'],
        'message': !exists(json, 'message') ? undefined : json['message'],
        'applications': !exists(json, 'applications') ? undefined : (json['applications'].map(ApplicationsFromJSON)),
        'nextToken': !exists(json, 'next_token') ? undefined : json['next_token'],
    };
}
export function GetApplicationsResponseToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'code': value.code,
        'message': value.message,
        'applications': value.applications === undefined ? undefined : (value.applications.map(ApplicationsToJSON)),
        'next_token': value.nextToken,
    };
}
