// @kinde-oss/kinde-typescript-sdk/dist/models/GetEnvironmentFeatureFlagsResponse.js
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists, mapValues } from '../runtime.js';
import { GetOrganizationFeatureFlagsResponseFeatureFlagsValueFromJSON, GetOrganizationFeatureFlagsResponseFeatureFlagsValueToJSON, } from './GetOrganizationFeatureFlagsResponseFeatureFlagsValue.js';
/**
 * Check if a given object implements the GetEnvironmentFeatureFlagsResponse interface.
 */
export function instanceOfGetEnvironmentFeatureFlagsResponse(value) {
    var isInstance = true;
    return isInstance;
}
export function GetEnvironmentFeatureFlagsResponseFromJSON(json) {
    return GetEnvironmentFeatureFlagsResponseFromJSONTyped(json, false);
}
export function GetEnvironmentFeatureFlagsResponseFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'code': !exists(json, 'code') ? undefined : json['code'],
        'message': !exists(json, 'message') ? undefined : json['message'],
        'featureFlags': !exists(json, 'feature_flags') ? undefined : (mapValues(json['feature_flags'], GetOrganizationFeatureFlagsResponseFeatureFlagsValueFromJSON)),
        'nextToken': !exists(json, 'next_token') ? undefined : json['next_token'],
    };
}
export function GetEnvironmentFeatureFlagsResponseToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'code': value.code,
        'message': value.message,
        'feature_flags': value.featureFlags === undefined ? undefined : (mapValues(value.featureFlags, GetOrganizationFeatureFlagsResponseFeatureFlagsValueToJSON)),
        'next_token': value.nextToken,
    };
}
