// @kinde-oss/kinde-typescript-sdk/dist/models/GetOrganizationsUserPermissionsResponse.js
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime.js';
import { OrganizationUserPermissionFromJSON, OrganizationUserPermissionToJSON, } from './OrganizationUserPermission.js';
/**
 * Check if a given object implements the GetOrganizationsUserPermissionsResponse interface.
 */
export function instanceOfGetOrganizationsUserPermissionsResponse(value) {
    var isInstance = true;
    return isInstance;
}
export function GetOrganizationsUserPermissionsResponseFromJSON(json) {
    return GetOrganizationsUserPermissionsResponseFromJSONTyped(json, false);
}
export function GetOrganizationsUserPermissionsResponseFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'code': !exists(json, 'code') ? undefined : json['code'],
        'message': !exists(json, 'message') ? undefined : json['message'],
        'permissions': !exists(json, 'permissions') ? undefined : (json['permissions'].map(OrganizationUserPermissionFromJSON)),
    };
}
export function GetOrganizationsUserPermissionsResponseToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'code': value.code,
        'message': value.message,
        'permissions': value.permissions === undefined ? undefined : (value.permissions.map(OrganizationUserPermissionToJSON)),
    };
}
