// @kinde-oss/kinde-typescript-sdk/dist/models/GetOrganizationFeatureFlagsResponseFeatureFlagsValue.js
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime.js';
/**
 * @export
 */
export var GetOrganizationFeatureFlagsResponseFeatureFlagsValueTypeEnum = {
    Str: 'str',
    Int: 'int',
    Bool: 'bool'
};
/**
 * Check if a given object implements the GetOrganizationFeatureFlagsResponseFeatureFlagsValue interface.
 */
export function instanceOfGetOrganizationFeatureFlagsResponseFeatureFlagsValue(value) {
    var isInstance = true;
    return isInstance;
}
export function GetOrganizationFeatureFlagsResponseFeatureFlagsValueFromJSON(json) {
    return GetOrganizationFeatureFlagsResponseFeatureFlagsValueFromJSONTyped(json, false);
}
export function GetOrganizationFeatureFlagsResponseFeatureFlagsValueFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'type': !exists(json, 'type') ? undefined : json['type'],
        'value': !exists(json, 'value') ? undefined : json['value'],
    };
}
export function GetOrganizationFeatureFlagsResponseFeatureFlagsValueToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'type': value.type,
        'value': value.value,
    };
}
