// @kinde-oss/kinde-typescript-sdk/dist/models/GetOrganizationsUserRolesResponse.js
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime.js';
import { OrganizationUserRoleFromJSON, OrganizationUserRoleToJSON, } from './OrganizationUserRole.js';
/**
 * Check if a given object implements the GetOrganizationsUserRolesResponse interface.
 */
export function instanceOfGetOrganizationsUserRolesResponse(value) {
    var isInstance = true;
    return isInstance;
}
export function GetOrganizationsUserRolesResponseFromJSON(json) {
    return GetOrganizationsUserRolesResponseFromJSONTyped(json, false);
}
export function GetOrganizationsUserRolesResponseFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'code': !exists(json, 'code') ? undefined : json['code'],
        'message': !exists(json, 'message') ? undefined : json['message'],
        'roles': !exists(json, 'roles') ? undefined : (json['roles'].map(OrganizationUserRoleFromJSON)),
        'nextToken': !exists(json, 'next_token') ? undefined : json['next_token'],
    };
}
export function GetOrganizationsUserRolesResponseToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'code': value.code,
        'message': value.message,
        'roles': value.roles === undefined ? undefined : (value.roles.map(OrganizationUserRoleToJSON)),
        'next_token': value.nextToken,
    };
}
