// @kinde-oss/kinde-typescript-sdk/dist/models/GetOrganizationUsersResponse.js
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime.js';
import { OrganizationUserFromJSON, OrganizationUserToJSON, } from './OrganizationUser.js';
/**
 * Check if a given object implements the GetOrganizationUsersResponse interface.
 */
export function instanceOfGetOrganizationUsersResponse(value) {
    var isInstance = true;
    return isInstance;
}
export function GetOrganizationUsersResponseFromJSON(json) {
    return GetOrganizationUsersResponseFromJSONTyped(json, false);
}
export function GetOrganizationUsersResponseFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'code': !exists(json, 'code') ? undefined : json['code'],
        'message': !exists(json, 'message') ? undefined : json['message'],
        'organizationUsers': !exists(json, 'organization_users') ? undefined : (json['organization_users'].map(OrganizationUserFromJSON)),
        'nextToken': !exists(json, 'next_token') ? undefined : json['next_token'],
    };
}
export function GetOrganizationUsersResponseToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'code': value.code,
        'message': value.message,
        'organization_users': value.organizationUsers === undefined ? undefined : (value.organizationUsers.map(OrganizationUserToJSON)),
        'next_token': value.nextToken,
    };
}
