// @kinde-oss/kinde-typescript-sdk/dist/models/GetOrganizationsResponse.js
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime.js';
import { OrganizationFromJSON, OrganizationToJSON, } from './Organization.js';
/**
 * Check if a given object implements the GetOrganizationsResponse interface.
 */
export function instanceOfGetOrganizationsResponse(value) {
    var isInstance = true;
    return isInstance;
}
export function GetOrganizationsResponseFromJSON(json) {
    return GetOrganizationsResponseFromJSONTyped(json, false);
}
export function GetOrganizationsResponseFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'code': !exists(json, 'code') ? undefined : json['code'],
        'message': !exists(json, 'message') ? undefined : json['message'],
        'organizations': !exists(json, 'organizations') ? undefined : (json['organizations'].map(OrganizationFromJSON)),
        'nextToken': !exists(json, 'next_token') ? undefined : json['next_token'],
    };
}
export function GetOrganizationsResponseToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'code': value.code,
        'message': value.message,
        'organizations': value.organizations === undefined ? undefined : (value.organizations.map(OrganizationToJSON)),
        'next_token': value.nextToken,
    };
}
