// @kinde-oss/kinde-typescript-sdk/dist/models/GetPermissionsResponse.js
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime.js';
import { PermissionsFromJSON, PermissionsToJSON, } from './Permissions.js';
/**
 * Check if a given object implements the GetPermissionsResponse interface.
 */
export function instanceOfGetPermissionsResponse(value) {
    var isInstance = true;
    return isInstance;
}
export function GetPermissionsResponseFromJSON(json) {
    return GetPermissionsResponseFromJSONTyped(json, false);
}
export function GetPermissionsResponseFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'code': !exists(json, 'code') ? undefined : json['code'],
        'message': !exists(json, 'message') ? undefined : json['message'],
        'permissions': !exists(json, 'permissions') ? undefined : (json['permissions'].map(PermissionsFromJSON)),
        'nextToken': !exists(json, 'next_token') ? undefined : json['next_token'],
    };
}
export function GetPermissionsResponseToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'code': value.code,
        'message': value.message,
        'permissions': value.permissions === undefined ? undefined : (value.permissions.map(PermissionsToJSON)),
        'next_token': value.nextToken,
    };
}
