// @kinde-oss/kinde-typescript-sdk/dist/models/GetRedirectCallbackUrlsResponse.js
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime.js';
import { RedirectCallbackUrlsFromJSON, RedirectCallbackUrlsToJSON, } from './RedirectCallbackUrls.js';
/**
 * Check if a given object implements the GetRedirectCallbackUrlsResponse interface.
 */
export function instanceOfGetRedirectCallbackUrlsResponse(value) {
    var isInstance = true;
    return isInstance;
}
export function GetRedirectCallbackUrlsResponseFromJSON(json) {
    return GetRedirectCallbackUrlsResponseFromJSONTyped(json, false);
}
export function GetRedirectCallbackUrlsResponseFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'redirectUrls': !exists(json, 'redirect_urls') ? undefined : (json['redirect_urls'].map(RedirectCallbackUrlsFromJSON)),
    };
}
export function GetRedirectCallbackUrlsResponseToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'redirect_urls': value.redirectUrls === undefined ? undefined : (value.redirectUrls.map(RedirectCallbackUrlsToJSON)),
    };
}
