// @kinde-oss/kinde-typescript-sdk/dist/models/GetSubscriberResponse.js
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime.js';
import { SubscriberFromJSON, SubscriberToJSON, } from './Subscriber.js';
/**
 * Check if a given object implements the GetSubscriberResponse interface.
 */
export function instanceOfGetSubscriberResponse(value) {
    var isInstance = true;
    return isInstance;
}
export function GetSubscriberResponseFromJSON(json) {
    return GetSubscriberResponseFromJSONTyped(json, false);
}
export function GetSubscriberResponseFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'code': !exists(json, 'code') ? undefined : json['code'],
        'message': !exists(json, 'message') ? undefined : json['message'],
        'subscribers': !exists(json, 'subscribers') ? undefined : (json['subscribers'].map(SubscriberFromJSON)),
    };
}
export function GetSubscriberResponseToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'code': value.code,
        'message': value.message,
        'subscribers': value.subscribers === undefined ? undefined : (value.subscribers.map(SubscriberToJSON)),
    };
}
