// @kinde-oss/kinde-typescript-sdk/dist/models/GetSubscribersResponse.js
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime.js';
import { SubscribersSubscriberFromJSON, SubscribersSubscriberToJSON, } from './SubscribersSubscriber.js';
/**
 * Check if a given object implements the GetSubscribersResponse interface.
 */
export function instanceOfGetSubscribersResponse(value) {
    var isInstance = true;
    return isInstance;
}
export function GetSubscribersResponseFromJSON(json) {
    return GetSubscribersResponseFromJSONTyped(json, false);
}
export function GetSubscribersResponseFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'code': !exists(json, 'code') ? undefined : json['code'],
        'message': !exists(json, 'message') ? undefined : json['message'],
        'subscribers': !exists(json, 'subscribers') ? undefined : (json['subscribers'].map(SubscribersSubscriberFromJSON)),
        'nextToken': !exists(json, 'next_token') ? undefined : json['next_token'],
    };
}
export function GetSubscribersResponseToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'code': value.code,
        'message': value.message,
        'subscribers': value.subscribers === undefined ? undefined : (value.subscribers.map(SubscribersSubscriberToJSON)),
        'next_token': value.nextToken,
    };
}
