// @kinde-oss/kinde-typescript-sdk/dist/models/LogoutRedirectUrls.js
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime.js';
/**
 * Check if a given object implements the LogoutRedirectUrls interface.
 */
export function instanceOfLogoutRedirectUrls(value) {
    var isInstance = true;
    return isInstance;
}
export function LogoutRedirectUrlsFromJSON(json) {
    return LogoutRedirectUrlsFromJSONTyped(json, false);
}
export function LogoutRedirectUrlsFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'redirectUrls': !exists(json, 'redirect_urls') ? undefined : json['redirect_urls'],
    };
}
export function LogoutRedirectUrlsToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'redirect_urls': value.redirectUrls,
    };
}
