// @kinde-oss/kinde-typescript-sdk/dist/models/OrganizationUserRolePermissions.js
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime.js';
import { OrganizationUserRolePermissionsPermissionsFromJSON, OrganizationUserRolePermissionsPermissionsToJSON, } from './OrganizationUserRolePermissionsPermissions.js';
/**
 * Check if a given object implements the OrganizationUserRolePermissions interface.
 */
export function instanceOfOrganizationUserRolePermissions(value) {
    var isInstance = true;
    return isInstance;
}
export function OrganizationUserRolePermissionsFromJSON(json) {
    return OrganizationUserRolePermissionsFromJSONTyped(json, false);
}
export function OrganizationUserRolePermissionsFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'id': !exists(json, 'id') ? undefined : json['id'],
        'role': !exists(json, 'role') ? undefined : json['role'],
        'permissions': !exists(json, 'permissions') ? undefined : OrganizationUserRolePermissionsPermissionsFromJSON(json['permissions']),
    };
}
export function OrganizationUserRolePermissionsToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'id': value.id,
        'role': value.role,
        'permissions': OrganizationUserRolePermissionsPermissionsToJSON(value.permissions),
    };
}
