// @kinde-oss/kinde-typescript-sdk/dist/models/OrganizationUserPermissionRolesInner.js
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime.js';
/**
 * Check if a given object implements the OrganizationUserPermissionRolesInner interface.
 */
export function instanceOfOrganizationUserPermissionRolesInner(value) {
    var isInstance = true;
    return isInstance;
}
export function OrganizationUserPermissionRolesInnerFromJSON(json) {
    return OrganizationUserPermissionRolesInnerFromJSONTyped(json, false);
}
export function OrganizationUserPermissionRolesInnerFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'id': !exists(json, 'id') ? undefined : json['id'],
        'key': !exists(json, 'key') ? undefined : json['key'],
    };
}
export function OrganizationUserPermissionRolesInnerToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'id': value.id,
        'key': value.key,
    };
}
