// @kinde-oss/kinde-typescript-sdk/dist/models/OrganizationUserPermission.js
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime.js';
import { OrganizationUserPermissionRolesInnerFromJSON, OrganizationUserPermissionRolesInnerToJSON, } from './OrganizationUserPermissionRolesInner.js';
/**
 * Check if a given object implements the OrganizationUserPermission interface.
 */
export function instanceOfOrganizationUserPermission(value) {
    var isInstance = true;
    return isInstance;
}
export function OrganizationUserPermissionFromJSON(json) {
    return OrganizationUserPermissionFromJSONTyped(json, false);
}
export function OrganizationUserPermissionFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'id': !exists(json, 'id') ? undefined : json['id'],
        'key': !exists(json, 'key') ? undefined : json['key'],
        'name': !exists(json, 'name') ? undefined : json['name'],
        'description': !exists(json, 'description') ? undefined : json['description'],
        'roles': !exists(json, 'roles') ? undefined : (json['roles'].map(OrganizationUserPermissionRolesInnerFromJSON)),
    };
}
export function OrganizationUserPermissionToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'id': value.id,
        'key': value.key,
        'name': value.name,
        'description': value.description,
        'roles': value.roles === undefined ? undefined : (value.roles.map(OrganizationUserPermissionRolesInnerToJSON)),
    };
}
