// @kinde-oss/kinde-typescript-sdk/dist/models/Organization.js
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime.js';
/**
 * Check if a given object implements the Organization interface.
 */
export function instanceOfOrganization(value) {
    var isInstance = true;
    return isInstance;
}
export function OrganizationFromJSON(json) {
    return OrganizationFromJSONTyped(json, false);
}
export function OrganizationFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'code': !exists(json, 'code') ? undefined : json['code'],
        'name': !exists(json, 'name') ? undefined : json['name'],
        'isDefault': !exists(json, 'is_default') ? undefined : json['is_default'],
        'externalId': !exists(json, 'external_id') ? undefined : json['external_id'],
    };
}
export function OrganizationToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'code': value.code,
        'name': value.name,
        'is_default': value.isDefault,
        'external_id': value.externalId,
    };
}
