// @kinde-oss/kinde-typescript-sdk/dist/models/ReplaceLogoutRedirectURLsRequest.js
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime.js';
/**
 * Check if a given object implements the ReplaceLogoutRedirectURLsRequest interface.
 */
export function instanceOfReplaceLogoutRedirectURLsRequest(value) {
    var isInstance = true;
    return isInstance;
}
export function ReplaceLogoutRedirectURLsRequestFromJSON(json) {
    return ReplaceLogoutRedirectURLsRequestFromJSONTyped(json, false);
}
export function ReplaceLogoutRedirectURLsRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'urls': !exists(json, 'urls') ? undefined : json['urls'],
    };
}
export function ReplaceLogoutRedirectURLsRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'urls': value.urls,
    };
}
