// @kinde-oss/kinde-typescript-sdk/dist/models/SubscribersSubscriber.js
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime.js';
/**
 * Check if a given object implements the SubscribersSubscriber interface.
 */
export function instanceOfSubscribersSubscriber(value) {
    var isInstance = true;
    return isInstance;
}
export function SubscribersSubscriberFromJSON(json) {
    return SubscribersSubscriberFromJSONTyped(json, false);
}
export function SubscribersSubscriberFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'id': !exists(json, 'id') ? undefined : json['id'],
        'email': !exists(json, 'email') ? undefined : json['email'],
        'fullName': !exists(json, 'full_name') ? undefined : json['full_name'],
        'firstName': !exists(json, 'first_name') ? undefined : json['first_name'],
        'lastName': !exists(json, 'last_name') ? undefined : json['last_name'],
    };
}
export function SubscribersSubscriberToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'id': value.id,
        'email': value.email,
        'full_name': value.fullName,
        'first_name': value.firstName,
        'last_name': value.lastName,
    };
}
