// @kinde-oss/kinde-typescript-sdk/dist/models/Subscriber.js
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime.js';
/**
 * Check if a given object implements the Subscriber interface.
 */
export function instanceOfSubscriber(value) {
    var isInstance = true;
    return isInstance;
}
export function SubscriberFromJSON(json) {
    return SubscriberFromJSONTyped(json, false);
}
export function SubscriberFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'id': !exists(json, 'id') ? undefined : json['id'],
        'preferredEmail': !exists(json, 'preferred_email') ? undefined : json['preferred_email'],
        'firstName': !exists(json, 'first_name') ? undefined : json['first_name'],
        'lastName': !exists(json, 'last_name') ? undefined : json['last_name'],
    };
}
export function SubscriberToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'id': value.id,
        'preferred_email': value.preferredEmail,
        'first_name': value.firstName,
        'last_name': value.lastName,
    };
}
