// @kinde-oss/kinde-typescript-sdk/dist/models/TokenIntrospect.js
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime.js';
/**
 * Check if a given object implements the TokenIntrospect interface.
 */
export function instanceOfTokenIntrospect(value) {
    var isInstance = true;
    return isInstance;
}
export function TokenIntrospectFromJSON(json) {
    return TokenIntrospectFromJSONTyped(json, false);
}
export function TokenIntrospectFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'active': !exists(json, 'active') ? undefined : json['active'],
        'aud': !exists(json, 'aud') ? undefined : json['aud'],
        'clientId': !exists(json, 'client_id') ? undefined : json['client_id'],
        'exp': !exists(json, 'exp') ? undefined : json['exp'],
        'iat': !exists(json, 'iat') ? undefined : json['iat'],
    };
}
export function TokenIntrospectToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'active': value.active,
        'aud': value.aud,
        'client_id': value.clientId,
        'exp': value.exp,
        'iat': value.iat,
    };
}
