// @kinde-oss/kinde-typescript-sdk/dist/models/TokenErrorResponse.js
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime.js';
/**
 * Check if a given object implements the TokenErrorResponse interface.
 */
export function instanceOfTokenErrorResponse(value) {
    var isInstance = true;
    return isInstance;
}
export function TokenErrorResponseFromJSON(json) {
    return TokenErrorResponseFromJSONTyped(json, false);
}
export function TokenErrorResponseFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'error': !exists(json, 'error') ? undefined : json['error'],
        'errorDescription': !exists(json, 'error_description') ? undefined : json['error_description'],
    };
}
export function TokenErrorResponseToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'error': value.error,
        'error_description': value.errorDescription,
    };
}
