// @kinde-oss/kinde-typescript-sdk/dist/models/UpdateUserRequest.js
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime.js';
/**
 * Check if a given object implements the UpdateUserRequest interface.
 */
export function instanceOfUpdateUserRequest(value) {
    var isInstance = true;
    return isInstance;
}
export function UpdateUserRequestFromJSON(json) {
    return UpdateUserRequestFromJSONTyped(json, false);
}
export function UpdateUserRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'givenName': !exists(json, 'given_name') ? undefined : json['given_name'],
        'familyName': !exists(json, 'family_name') ? undefined : json['family_name'],
        'isSuspended': !exists(json, 'is_suspended') ? undefined : json['is_suspended'],
        'isPasswordResetRequested': !exists(json, 'is_password_reset_requested') ? undefined : json['is_password_reset_requested'],
    };
}
export function UpdateUserRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'given_name': value.givenName,
        'family_name': value.familyName,
        'is_suspended': value.isSuspended,
        'is_password_reset_requested': value.isPasswordResetRequested,
    };
}
