// @kinde-oss/kinde-typescript-sdk/dist/models/UpdateRolePermissionsResponse.js
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime.js';
/**
 * Check if a given object implements the UpdateRolePermissionsResponse interface.
 */
export function instanceOfUpdateRolePermissionsResponse(value) {
    var isInstance = true;
    return isInstance;
}
export function UpdateRolePermissionsResponseFromJSON(json) {
    return UpdateRolePermissionsResponseFromJSONTyped(json, false);
}
export function UpdateRolePermissionsResponseFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'code': !exists(json, 'code') ? undefined : json['code'],
        'message': !exists(json, 'message') ? undefined : json['message'],
        'permissionsAdded': !exists(json, 'permissions_added') ? undefined : json['permissions_added'],
        'permissionsRemoved': !exists(json, 'permissions_removed') ? undefined : json['permissions_removed'],
    };
}
export function UpdateRolePermissionsResponseToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'code': value.code,
        'message': value.message,
        'permissions_added': value.permissionsAdded,
        'permissions_removed': value.permissionsRemoved,
    };
}
