// @kinde-oss/kinde-typescript-sdk/dist/models/UpdateOrganizationUsersRequest.js
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime.js';
import { UpdateOrganizationUsersRequestUsersInnerFromJSON, UpdateOrganizationUsersRequestUsersInnerToJSON, } from './UpdateOrganizationUsersRequestUsersInner.js';
/**
 * Check if a given object implements the UpdateOrganizationUsersRequest interface.
 */
export function instanceOfUpdateOrganizationUsersRequest(value) {
    var isInstance = true;
    return isInstance;
}
export function UpdateOrganizationUsersRequestFromJSON(json) {
    return UpdateOrganizationUsersRequestFromJSONTyped(json, false);
}
export function UpdateOrganizationUsersRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'users': !exists(json, 'users') ? undefined : (json['users'].map(UpdateOrganizationUsersRequestUsersInnerFromJSON)),
    };
}
export function UpdateOrganizationUsersRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'users': value.users === undefined ? undefined : (value.users.map(UpdateOrganizationUsersRequestUsersInnerToJSON)),
    };
}
