// @kinde-oss/kinde-typescript-sdk/dist/models/UpdateApplicationRequest.js
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime.js';
/**
 * Check if a given object implements the UpdateApplicationRequest interface.
 */
export function instanceOfUpdateApplicationRequest(value) {
    var isInstance = true;
    return isInstance;
}
export function UpdateApplicationRequestFromJSON(json) {
    return UpdateApplicationRequestFromJSONTyped(json, false);
}
export function UpdateApplicationRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': !exists(json, 'name') ? undefined : json['name'],
        'languageKey': !exists(json, 'language_key') ? undefined : json['language_key'],
        'logoutUris': !exists(json, 'logout_uris') ? undefined : json['logout_uris'],
        'redirectUris': !exists(json, 'redirect_uris') ? undefined : json['redirect_uris'],
    };
}
export function UpdateApplicationRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'language_key': value.languageKey,
        'logout_uris': value.logoutUris,
        'redirect_uris': value.redirectUris,
    };
}
