// @kinde-oss/kinde-typescript-sdk/dist/models/UpdateOrganizationUsersRequestUsersInner.js
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime.js';
/**
 * Check if a given object implements the UpdateOrganizationUsersRequestUsersInner interface.
 */
export function instanceOfUpdateOrganizationUsersRequestUsersInner(value) {
    var isInstance = true;
    return isInstance;
}
export function UpdateOrganizationUsersRequestUsersInnerFromJSON(json) {
    return UpdateOrganizationUsersRequestUsersInnerFromJSONTyped(json, false);
}
export function UpdateOrganizationUsersRequestUsersInnerFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'id': !exists(json, 'id') ? undefined : json['id'],
        'operation': !exists(json, 'operation') ? undefined : json['operation'],
        'roles': !exists(json, 'roles') ? undefined : json['roles'],
        'permissions': !exists(json, 'permissions') ? undefined : json['permissions'],
    };
}
export function UpdateOrganizationUsersRequestUsersInnerToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'id': value.id,
        'operation': value.operation,
        'roles': value.roles,
        'permissions': value.permissions,
    };
}
