// @kinde-oss/kinde-typescript-sdk/dist/models/UpdateUserResponse.js
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime.js';
/**
 * Check if a given object implements the UpdateUserResponse interface.
 */
export function instanceOfUpdateUserResponse(value) {
    var isInstance = true;
    return isInstance;
}
export function UpdateUserResponseFromJSON(json) {
    return UpdateUserResponseFromJSONTyped(json, false);
}
export function UpdateUserResponseFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'id': !exists(json, 'id') ? undefined : json['id'],
        'givenName': !exists(json, 'given_name') ? undefined : json['given_name'],
        'familyName': !exists(json, 'family_name') ? undefined : json['family_name'],
        'email': !exists(json, 'email') ? undefined : json['email'],
        'isSuspended': !exists(json, 'is_suspended') ? undefined : json['is_suspended'],
        'isPasswordResetRequested': !exists(json, 'is_password_reset_requested') ? undefined : json['is_password_reset_requested'],
        'picture': !exists(json, 'picture') ? undefined : json['picture'],
    };
}
export function UpdateUserResponseToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'id': value.id,
        'given_name': value.givenName,
        'family_name': value.familyName,
        'email': value.email,
        'is_suspended': value.isSuspended,
        'is_password_reset_requested': value.isPasswordResetRequested,
        'picture': value.picture,
    };
}
