// @kinde-oss/kinde-typescript-sdk/dist/models/UpdateOrganizationUsersResponse.js
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime.js';
/**
 * Check if a given object implements the UpdateOrganizationUsersResponse interface.
 */
export function instanceOfUpdateOrganizationUsersResponse(value) {
    var isInstance = true;
    return isInstance;
}
export function UpdateOrganizationUsersResponseFromJSON(json) {
    return UpdateOrganizationUsersResponseFromJSONTyped(json, false);
}
export function UpdateOrganizationUsersResponseFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'message': !exists(json, 'message') ? undefined : json['message'],
        'usersAdded': !exists(json, 'users_added') ? undefined : json['users_added'],
        'usersUpdated': !exists(json, 'users_updated') ? undefined : json['users_updated'],
        'usersRemoved': !exists(json, 'users_removed') ? undefined : json['users_removed'],
    };
}
export function UpdateOrganizationUsersResponseToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'message': value.message,
        'users_added': value.usersAdded,
        'users_updated': value.usersUpdated,
        'users_removed': value.usersRemoved,
    };
}
