// @kinde-oss/kinde-typescript-sdk/dist/models/UpdateAPIApplicationsRequest.js
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { UpdateAPIApplicationsRequestApplicationsInnerFromJSON, UpdateAPIApplicationsRequestApplicationsInnerToJSON, } from './UpdateAPIApplicationsRequestApplicationsInner.js';
/**
 * Check if a given object implements the UpdateAPIApplicationsRequest interface.
 */
export function instanceOfUpdateAPIApplicationsRequest(value) {
    var isInstance = true;
    isInstance = isInstance && "applications" in value;
    return isInstance;
}
export function UpdateAPIApplicationsRequestFromJSON(json) {
    return UpdateAPIApplicationsRequestFromJSONTyped(json, false);
}
export function UpdateAPIApplicationsRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'applications': (json['applications'].map(UpdateAPIApplicationsRequestApplicationsInnerFromJSON)),
    };
}
export function UpdateAPIApplicationsRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'applications': (value.applications.map(UpdateAPIApplicationsRequestApplicationsInnerToJSON)),
    };
}
