// @kinde-oss/kinde-typescript-sdk/dist/models/UpdateRolesRequest.js
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime.js';
/**
 * Check if a given object implements the UpdateRolesRequest interface.
 */
export function instanceOfUpdateRolesRequest(value) {
    var isInstance = true;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "key" in value;
    return isInstance;
}
export function UpdateRolesRequestFromJSON(json) {
    return UpdateRolesRequestFromJSONTyped(json, false);
}
export function UpdateRolesRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': json['name'],
        'description': !exists(json, 'description') ? undefined : json['description'],
        'key': json['key'],
        'isDefaultRole': !exists(json, 'is_default_role') ? undefined : json['is_default_role'],
    };
}
export function UpdateRolesRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'description': value.description,
        'key': value.key,
        'is_default_role': value.isDefaultRole,
    };
}
