// @kinde-oss/kinde-typescript-sdk/dist/models/UpdateRolePermissionsRequest.js
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime.js';
import { UpdateRolePermissionsRequestPermissionsInnerFromJSON, UpdateRolePermissionsRequestPermissionsInnerToJSON, } from './UpdateRolePermissionsRequestPermissionsInner.js';
/**
 * Check if a given object implements the UpdateRolePermissionsRequest interface.
 */
export function instanceOfUpdateRolePermissionsRequest(value) {
    var isInstance = true;
    return isInstance;
}
export function UpdateRolePermissionsRequestFromJSON(json) {
    return UpdateRolePermissionsRequestFromJSONTyped(json, false);
}
export function UpdateRolePermissionsRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'permissions': !exists(json, 'permissions') ? undefined : (json['permissions'].map(UpdateRolePermissionsRequestPermissionsInnerFromJSON)),
    };
}
export function UpdateRolePermissionsRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'permissions': value.permissions === undefined ? undefined : (value.permissions.map(UpdateRolePermissionsRequestPermissionsInnerToJSON)),
    };
}
