// @kinde-oss/kinde-typescript-sdk/dist/models/UpdateEnvironementFeatureFlagOverrideRequest.js
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
/**
 * Check if a given object implements the UpdateEnvironementFeatureFlagOverrideRequest interface.
 */
export function instanceOfUpdateEnvironementFeatureFlagOverrideRequest(value) {
    var isInstance = true;
    isInstance = isInstance && "value" in value;
    return isInstance;
}
export function UpdateEnvironementFeatureFlagOverrideRequestFromJSON(json) {
    return UpdateEnvironementFeatureFlagOverrideRequestFromJSONTyped(json, false);
}
export function UpdateEnvironementFeatureFlagOverrideRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'value': json['value'],
    };
}
export function UpdateEnvironementFeatureFlagOverrideRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'value': value.value,
    };
}
