// @kinde-oss/kinde-typescript-sdk/dist/models/UserProfileV2.js
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime.js';
/**
 * Check if a given object implements the UserProfileV2 interface.
 */
export function instanceOfUserProfileV2(value) {
    var isInstance = true;
    return isInstance;
}
export function UserProfileV2FromJSON(json) {
    return UserProfileV2FromJSONTyped(json, false);
}
export function UserProfileV2FromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'id': !exists(json, 'id') ? undefined : json['id'],
        'sub': !exists(json, 'sub') ? undefined : json['sub'],
        'providedId': !exists(json, 'provided_id') ? undefined : json['provided_id'],
        'name': !exists(json, 'name') ? undefined : json['name'],
        'givenName': !exists(json, 'given_name') ? undefined : json['given_name'],
        'familyName': !exists(json, 'family_name') ? undefined : json['family_name'],
        'updatedAt': !exists(json, 'updated_at') ? undefined : json['updated_at'],
        'email': !exists(json, 'email') ? undefined : json['email'],
        'picture': !exists(json, 'picture') ? undefined : json['picture'],
    };
}
export function UserProfileV2ToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'id': value.id,
        'sub': value.sub,
        'provided_id': value.providedId,
        'name': value.name,
        'given_name': value.givenName,
        'family_name': value.familyName,
        'updated_at': value.updatedAt,
        'email': value.email,
        'picture': value.picture,
    };
}
