// @kinde-oss/kinde-typescript-sdk/dist/models/UserIdentitiesInner.js
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime.js';
/**
 * Check if a given object implements the UserIdentitiesInner interface.
 */
export function instanceOfUserIdentitiesInner(value) {
    var isInstance = true;
    return isInstance;
}
export function UserIdentitiesInnerFromJSON(json) {
    return UserIdentitiesInnerFromJSONTyped(json, false);
}
export function UserIdentitiesInnerFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'type': !exists(json, 'type') ? undefined : json['type'],
        'identity': !exists(json, 'identity') ? undefined : json['identity'],
    };
}
export function UserIdentitiesInnerToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'type': value.type,
        'identity': value.identity,
    };
}
