// @kinde-oss/kinde-typescript-sdk/dist/models/UserIdentity.js
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime.js';
import { UserIdentityResultFromJSON, UserIdentityResultToJSON, } from './UserIdentityResult.js';
/**
 * Check if a given object implements the UserIdentity interface.
 */
export function instanceOfUserIdentity(value) {
    var isInstance = true;
    return isInstance;
}
export function UserIdentityFromJSON(json) {
    return UserIdentityFromJSONTyped(json, false);
}
export function UserIdentityFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'type': !exists(json, 'type') ? undefined : json['type'],
        'result': !exists(json, 'result') ? undefined : UserIdentityResultFromJSON(json['result']),
    };
}
export function UserIdentityToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'type': value.type,
        'result': UserIdentityResultToJSON(value.result),
    };
}
