// @kinde-oss/kinde-typescript-sdk/dist/models/UserIdentityResult.js
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime.js';
/**
 * Check if a given object implements the UserIdentityResult interface.
 */
export function instanceOfUserIdentityResult(value) {
    var isInstance = true;
    return isInstance;
}
export function UserIdentityResultFromJSON(json) {
    return UserIdentityResultFromJSONTyped(json, false);
}
export function UserIdentityResultFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'created': !exists(json, 'created') ? undefined : json['created'],
    };
}
export function UserIdentityResultToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'created': value.created,
    };
}
