// @kinde-oss/kinde-typescript-sdk/dist-cjs/models/AddAPIsRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AddAPIsRequestToJSON = exports.AddAPIsRequestFromJSONTyped = exports.AddAPIsRequestFromJSON = exports.instanceOfAddAPIsRequest = void 0;
/**
 * Check if a given object implements the AddAPIsRequest interface.
 */
function instanceOfAddAPIsRequest(value) {
    var isInstance = true;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "audience" in value;
    return isInstance;
}
exports.instanceOfAddAPIsRequest = instanceOfAddAPIsRequest;
function AddAPIsRequestFromJSON(json) {
    return AddAPIsRequestFromJSONTyped(json, false);
}
exports.AddAPIsRequestFromJSON = AddAPIsRequestFromJSON;
function AddAPIsRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': json['name'],
        'audience': json['audience'],
    };
}
exports.AddAPIsRequestFromJSONTyped = AddAPIsRequestFromJSONTyped;
function AddAPIsRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'audience': value.audience,
    };
}
exports.AddAPIsRequestToJSON = AddAPIsRequestToJSON;
