// @kinde-oss/kinde-typescript-sdk/dist-cjs/models/AddOrganizationUsersRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AddOrganizationUsersRequestToJSON = exports.AddOrganizationUsersRequestFromJSONTyped = exports.AddOrganizationUsersRequestFromJSON = exports.instanceOfAddOrganizationUsersRequest = void 0;
var runtime_js_1 = require("../runtime.js");
var AddOrganizationUsersRequestUsersInner_js_1 = require("./AddOrganizationUsersRequestUsersInner.js");
/**
 * Check if a given object implements the AddOrganizationUsersRequest interface.
 */
function instanceOfAddOrganizationUsersRequest(value) {
    var isInstance = true;
    return isInstance;
}
exports.instanceOfAddOrganizationUsersRequest = instanceOfAddOrganizationUsersRequest;
function AddOrganizationUsersRequestFromJSON(json) {
    return AddOrganizationUsersRequestFromJSONTyped(json, false);
}
exports.AddOrganizationUsersRequestFromJSON = AddOrganizationUsersRequestFromJSON;
function AddOrganizationUsersRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'users': !(0, runtime_js_1.exists)(json, 'users') ? undefined : (json['users'].map(AddOrganizationUsersRequestUsersInner_js_1.AddOrganizationUsersRequestUsersInnerFromJSON)),
    };
}
exports.AddOrganizationUsersRequestFromJSONTyped = AddOrganizationUsersRequestFromJSONTyped;
function AddOrganizationUsersRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'users': value.users === undefined ? undefined : (value.users.map(AddOrganizationUsersRequestUsersInner_js_1.AddOrganizationUsersRequestUsersInnerToJSON)),
    };
}
exports.AddOrganizationUsersRequestToJSON = AddOrganizationUsersRequestToJSON;
