// @kinde-oss/kinde-typescript-sdk/dist-cjs/models/AddOrganizationUsersRequestUsersInner.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AddOrganizationUsersRequestUsersInnerToJSON = exports.AddOrganizationUsersRequestUsersInnerFromJSONTyped = exports.AddOrganizationUsersRequestUsersInnerFromJSON = exports.instanceOfAddOrganizationUsersRequestUsersInner = void 0;
var runtime_js_1 = require("../runtime.js");
/**
 * Check if a given object implements the AddOrganizationUsersRequestUsersInner interface.
 */
function instanceOfAddOrganizationUsersRequestUsersInner(value) {
    var isInstance = true;
    return isInstance;
}
exports.instanceOfAddOrganizationUsersRequestUsersInner = instanceOfAddOrganizationUsersRequestUsersInner;
function AddOrganizationUsersRequestUsersInnerFromJSON(json) {
    return AddOrganizationUsersRequestUsersInnerFromJSONTyped(json, false);
}
exports.AddOrganizationUsersRequestUsersInnerFromJSON = AddOrganizationUsersRequestUsersInnerFromJSON;
function AddOrganizationUsersRequestUsersInnerFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'id': !(0, runtime_js_1.exists)(json, 'id') ? undefined : json['id'],
        'roles': !(0, runtime_js_1.exists)(json, 'roles') ? undefined : json['roles'],
        'permissions': !(0, runtime_js_1.exists)(json, 'permissions') ? undefined : json['permissions'],
    };
}
exports.AddOrganizationUsersRequestUsersInnerFromJSONTyped = AddOrganizationUsersRequestUsersInnerFromJSONTyped;
function AddOrganizationUsersRequestUsersInnerToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'id': value.id,
        'roles': value.roles,
        'permissions': value.permissions,
    };
}
exports.AddOrganizationUsersRequestUsersInnerToJSON = AddOrganizationUsersRequestUsersInnerToJSON;
