// @kinde-oss/kinde-typescript-sdk/dist-cjs/models/AddOrganizationUsersResponse.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AddOrganizationUsersResponseToJSON = exports.AddOrganizationUsersResponseFromJSONTyped = exports.AddOrganizationUsersResponseFromJSON = exports.instanceOfAddOrganizationUsersResponse = void 0;
var runtime_js_1 = require("../runtime.js");
/**
 * Check if a given object implements the AddOrganizationUsersResponse interface.
 */
function instanceOfAddOrganizationUsersResponse(value) {
    var isInstance = true;
    return isInstance;
}
exports.instanceOfAddOrganizationUsersResponse = instanceOfAddOrganizationUsersResponse;
function AddOrganizationUsersResponseFromJSON(json) {
    return AddOrganizationUsersResponseFromJSONTyped(json, false);
}
exports.AddOrganizationUsersResponseFromJSON = AddOrganizationUsersResponseFromJSON;
function AddOrganizationUsersResponseFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'code': !(0, runtime_js_1.exists)(json, 'code') ? undefined : json['code'],
        'message': !(0, runtime_js_1.exists)(json, 'message') ? undefined : json['message'],
        'usersAdded': !(0, runtime_js_1.exists)(json, 'users_added') ? undefined : json['users_added'],
    };
}
exports.AddOrganizationUsersResponseFromJSONTyped = AddOrganizationUsersResponseFromJSONTyped;
function AddOrganizationUsersResponseToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'code': value.code,
        'message': value.message,
        'users_added': value.usersAdded,
    };
}
exports.AddOrganizationUsersResponseToJSON = AddOrganizationUsersResponseToJSON;
