// @kinde-oss/kinde-typescript-sdk/dist-cjs/models/Api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ApiToJSON = exports.ApiFromJSONTyped = exports.ApiFromJSON = exports.instanceOfApi = void 0;
var runtime_js_1 = require("../runtime.js");
var ApiApplicationsInner_js_1 = require("./ApiApplicationsInner.js");
/**
 * Check if a given object implements the Api interface.
 */
function instanceOfApi(value) {
    var isInstance = true;
    return isInstance;
}
exports.instanceOfApi = instanceOfApi;
function ApiFromJSON(json) {
    return ApiFromJSONTyped(json, false);
}
exports.ApiFromJSON = ApiFromJSON;
function ApiFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'id': !(0, runtime_js_1.exists)(json, 'id') ? undefined : json['id'],
        'code': !(0, runtime_js_1.exists)(json, 'code') ? undefined : json['code'],
        'name': !(0, runtime_js_1.exists)(json, 'name') ? undefined : json['name'],
        'message': !(0, runtime_js_1.exists)(json, 'message') ? undefined : json['message'],
        'audience': !(0, runtime_js_1.exists)(json, 'audience') ? undefined : json['audience'],
        'applications': !(0, runtime_js_1.exists)(json, 'applications') ? undefined : (json['applications'].map(ApiApplicationsInner_js_1.ApiApplicationsInnerFromJSON)),
    };
}
exports.ApiFromJSONTyped = ApiFromJSONTyped;
function ApiToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'id': value.id,
        'code': value.code,
        'name': value.name,
        'message': value.message,
        'audience': value.audience,
        'applications': value.applications === undefined ? undefined : (value.applications.map(ApiApplicationsInner_js_1.ApiApplicationsInnerToJSON)),
    };
}
exports.ApiToJSON = ApiToJSON;
