// @kinde-oss/kinde-typescript-sdk/dist-cjs/models/ApiApplicationsInner.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ApiApplicationsInnerToJSON = exports.ApiApplicationsInnerFromJSONTyped = exports.ApiApplicationsInnerFromJSON = exports.instanceOfApiApplicationsInner = void 0;
var runtime_js_1 = require("../runtime.js");
/**
 * Check if a given object implements the ApiApplicationsInner interface.
 */
function instanceOfApiApplicationsInner(value) {
    var isInstance = true;
    return isInstance;
}
exports.instanceOfApiApplicationsInner = instanceOfApiApplicationsInner;
function ApiApplicationsInnerFromJSON(json) {
    return ApiApplicationsInnerFromJSONTyped(json, false);
}
exports.ApiApplicationsInnerFromJSON = ApiApplicationsInnerFromJSON;
function ApiApplicationsInnerFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'id': !(0, runtime_js_1.exists)(json, 'id') ? undefined : json['id'],
        'name': !(0, runtime_js_1.exists)(json, 'name') ? undefined : json['name'],
        'type': !(0, runtime_js_1.exists)(json, 'type') ? undefined : json['type'],
        'isActive': !(0, runtime_js_1.exists)(json, 'is_active') ? undefined : json['is_active'],
    };
}
exports.ApiApplicationsInnerFromJSONTyped = ApiApplicationsInnerFromJSONTyped;
function ApiApplicationsInnerToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'id': value.id,
        'name': value.name,
        'type': value.type,
        'is_active': value.isActive,
    };
}
exports.ApiApplicationsInnerToJSON = ApiApplicationsInnerToJSON;
