// @kinde-oss/kinde-typescript-sdk/dist-cjs/models/ApiResult.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ApiResultToJSON = exports.ApiResultFromJSONTyped = exports.ApiResultFromJSON = exports.instanceOfApiResult = void 0;
var runtime_js_1 = require("../runtime.js");
/**
 * Check if a given object implements the ApiResult interface.
 */
function instanceOfApiResult(value) {
    var isInstance = true;
    return isInstance;
}
exports.instanceOfApiResult = instanceOfApiResult;
function ApiResultFromJSON(json) {
    return ApiResultFromJSONTyped(json, false);
}
exports.ApiResultFromJSON = ApiResultFromJSON;
function ApiResultFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'result': !(0, runtime_js_1.exists)(json, 'result') ? undefined : json['result'],
    };
}
exports.ApiResultFromJSONTyped = ApiResultFromJSONTyped;
function ApiResultToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'result': value.result,
    };
}
exports.ApiResultToJSON = ApiResultToJSON;
