// @kinde-oss/kinde-typescript-sdk/dist-cjs/models/Apis.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ApisToJSON = exports.ApisFromJSONTyped = exports.ApisFromJSON = exports.instanceOfApis = void 0;
var runtime_js_1 = require("../runtime.js");
/**
 * Check if a given object implements the Apis interface.
 */
function instanceOfApis(value) {
    var isInstance = true;
    return isInstance;
}
exports.instanceOfApis = instanceOfApis;
function ApisFromJSON(json) {
    return ApisFromJSONTyped(json, false);
}
exports.ApisFromJSON = ApisFromJSON;
function ApisFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'id': !(0, runtime_js_1.exists)(json, 'id') ? undefined : json['id'],
        'name': !(0, runtime_js_1.exists)(json, 'name') ? undefined : json['name'],
        'audience': !(0, runtime_js_1.exists)(json, 'audience') ? undefined : json['audience'],
        'isManagementApi': !(0, runtime_js_1.exists)(json, 'is_management_api') ? undefined : json['is_management_api'],
    };
}
exports.ApisFromJSONTyped = ApisFromJSONTyped;
function ApisToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'id': value.id,
        'name': value.name,
        'audience': value.audience,
        'is_management_api': value.isManagementApi,
    };
}
exports.ApisToJSON = ApisToJSON;
