// @kinde-oss/kinde-typescript-sdk/dist-cjs/models/Applications.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ApplicationsToJSON = exports.ApplicationsFromJSONTyped = exports.ApplicationsFromJSON = exports.instanceOfApplications = void 0;
var runtime_js_1 = require("../runtime.js");
/**
 * Check if a given object implements the Applications interface.
 */
function instanceOfApplications(value) {
    var isInstance = true;
    return isInstance;
}
exports.instanceOfApplications = instanceOfApplications;
function ApplicationsFromJSON(json) {
    return ApplicationsFromJSONTyped(json, false);
}
exports.ApplicationsFromJSON = ApplicationsFromJSON;
function ApplicationsFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'id': !(0, runtime_js_1.exists)(json, 'id') ? undefined : json['id'],
        'name': !(0, runtime_js_1.exists)(json, 'name') ? undefined : json['name'],
        'type': !(0, runtime_js_1.exists)(json, 'type') ? undefined : json['type'],
    };
}
exports.ApplicationsFromJSONTyped = ApplicationsFromJSONTyped;
function ApplicationsToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'id': value.id,
        'name': value.name,
        'type': value.type,
    };
}
exports.ApplicationsToJSON = ApplicationsToJSON;
