// @kinde-oss/kinde-typescript-sdk/dist-cjs/models/ConnectedAppsAccessToken.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ConnectedAppsAccessTokenToJSON = exports.ConnectedAppsAccessTokenFromJSONTyped = exports.ConnectedAppsAccessTokenFromJSON = exports.instanceOfConnectedAppsAccessToken = void 0;
var runtime_js_1 = require("../runtime.js");
/**
 * Check if a given object implements the ConnectedAppsAccessToken interface.
 */
function instanceOfConnectedAppsAccessToken(value) {
    var isInstance = true;
    return isInstance;
}
exports.instanceOfConnectedAppsAccessToken = instanceOfConnectedAppsAccessToken;
function ConnectedAppsAccessTokenFromJSON(json) {
    return ConnectedAppsAccessTokenFromJSONTyped(json, false);
}
exports.ConnectedAppsAccessTokenFromJSON = ConnectedAppsAccessTokenFromJSON;
function ConnectedAppsAccessTokenFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'accessToken': !(0, runtime_js_1.exists)(json, 'access_token') ? undefined : json['access_token'],
        'accessTokenExpiry': !(0, runtime_js_1.exists)(json, 'access_token_expiry') ? undefined : json['access_token_expiry'],
    };
}
exports.ConnectedAppsAccessTokenFromJSONTyped = ConnectedAppsAccessTokenFromJSONTyped;
function ConnectedAppsAccessTokenToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'access_token': value.accessToken,
        'access_token_expiry': value.accessTokenExpiry,
    };
}
exports.ConnectedAppsAccessTokenToJSON = ConnectedAppsAccessTokenToJSON;
