// @kinde-oss/kinde-typescript-sdk/dist-cjs/models/ConnectedAppsAuthUrl.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ConnectedAppsAuthUrlToJSON = exports.ConnectedAppsAuthUrlFromJSONTyped = exports.ConnectedAppsAuthUrlFromJSON = exports.instanceOfConnectedAppsAuthUrl = void 0;
var runtime_js_1 = require("../runtime.js");
/**
 * Check if a given object implements the ConnectedAppsAuthUrl interface.
 */
function instanceOfConnectedAppsAuthUrl(value) {
    var isInstance = true;
    return isInstance;
}
exports.instanceOfConnectedAppsAuthUrl = instanceOfConnectedAppsAuthUrl;
function ConnectedAppsAuthUrlFromJSON(json) {
    return ConnectedAppsAuthUrlFromJSONTyped(json, false);
}
exports.ConnectedAppsAuthUrlFromJSON = ConnectedAppsAuthUrlFromJSON;
function ConnectedAppsAuthUrlFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'url': !(0, runtime_js_1.exists)(json, 'url') ? undefined : json['url'],
        'sessionId': !(0, runtime_js_1.exists)(json, 'session_id') ? undefined : json['session_id'],
    };
}
exports.ConnectedAppsAuthUrlFromJSONTyped = ConnectedAppsAuthUrlFromJSONTyped;
function ConnectedAppsAuthUrlToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'url': value.url,
        'session_id': value.sessionId,
    };
}
exports.ConnectedAppsAuthUrlToJSON = ConnectedAppsAuthUrlToJSON;
