// @kinde-oss/kinde-typescript-sdk/dist-cjs/models/CreateSubscriberSuccessResponse.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateSubscriberSuccessResponseToJSON = exports.CreateSubscriberSuccessResponseFromJSONTyped = exports.CreateSubscriberSuccessResponseFromJSON = exports.instanceOfCreateSubscriberSuccessResponse = void 0;
var runtime_js_1 = require("../runtime.js");
var CreateSubscriberSuccessResponseSubscriber_js_1 = require("./CreateSubscriberSuccessResponseSubscriber.js");
/**
 * Check if a given object implements the CreateSubscriberSuccessResponse interface.
 */
function instanceOfCreateSubscriberSuccessResponse(value) {
    var isInstance = true;
    return isInstance;
}
exports.instanceOfCreateSubscriberSuccessResponse = instanceOfCreateSubscriberSuccessResponse;
function CreateSubscriberSuccessResponseFromJSON(json) {
    return CreateSubscriberSuccessResponseFromJSONTyped(json, false);
}
exports.CreateSubscriberSuccessResponseFromJSON = CreateSubscriberSuccessResponseFromJSON;
function CreateSubscriberSuccessResponseFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'subscriber': !(0, runtime_js_1.exists)(json, 'subscriber') ? undefined : (0, CreateSubscriberSuccessResponseSubscriber_js_1.CreateSubscriberSuccessResponseSubscriberFromJSON)(json['subscriber']),
    };
}
exports.CreateSubscriberSuccessResponseFromJSONTyped = CreateSubscriberSuccessResponseFromJSONTyped;
function CreateSubscriberSuccessResponseToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'subscriber': (0, CreateSubscriberSuccessResponseSubscriber_js_1.CreateSubscriberSuccessResponseSubscriberToJSON)(value.subscriber),
    };
}
exports.CreateSubscriberSuccessResponseToJSON = CreateSubscriberSuccessResponseToJSON;
