// @kinde-oss/kinde-typescript-sdk/dist-cjs/models/CreateUserRequestProfile.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateUserRequestProfileToJSON = exports.CreateUserRequestProfileFromJSONTyped = exports.CreateUserRequestProfileFromJSON = exports.instanceOfCreateUserRequestProfile = void 0;
var runtime_js_1 = require("../runtime.js");
/**
 * Check if a given object implements the CreateUserRequestProfile interface.
 */
function instanceOfCreateUserRequestProfile(value) {
    var isInstance = true;
    return isInstance;
}
exports.instanceOfCreateUserRequestProfile = instanceOfCreateUserRequestProfile;
function CreateUserRequestProfileFromJSON(json) {
    return CreateUserRequestProfileFromJSONTyped(json, false);
}
exports.CreateUserRequestProfileFromJSON = CreateUserRequestProfileFromJSON;
function CreateUserRequestProfileFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'givenName': !(0, runtime_js_1.exists)(json, 'given_name') ? undefined : json['given_name'],
        'familyName': !(0, runtime_js_1.exists)(json, 'family_name') ? undefined : json['family_name'],
    };
}
exports.CreateUserRequestProfileFromJSONTyped = CreateUserRequestProfileFromJSONTyped;
function CreateUserRequestProfileToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'given_name': value.givenName,
        'family_name': value.familyName,
    };
}
exports.CreateUserRequestProfileToJSON = CreateUserRequestProfileToJSON;
