// @kinde-oss/kinde-typescript-sdk/dist-cjs/models/CreateOrganizationResponse.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateOrganizationResponseToJSON = exports.CreateOrganizationResponseFromJSONTyped = exports.CreateOrganizationResponseFromJSON = exports.instanceOfCreateOrganizationResponse = void 0;
var runtime_js_1 = require("../runtime.js");
var CreateOrganizationResponseOrganization_js_1 = require("./CreateOrganizationResponseOrganization.js");
/**
 * Check if a given object implements the CreateOrganizationResponse interface.
 */
function instanceOfCreateOrganizationResponse(value) {
    var isInstance = true;
    return isInstance;
}
exports.instanceOfCreateOrganizationResponse = instanceOfCreateOrganizationResponse;
function CreateOrganizationResponseFromJSON(json) {
    return CreateOrganizationResponseFromJSONTyped(json, false);
}
exports.CreateOrganizationResponseFromJSON = CreateOrganizationResponseFromJSON;
function CreateOrganizationResponseFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'message': !(0, runtime_js_1.exists)(json, 'message') ? undefined : json['message'],
        'code': !(0, runtime_js_1.exists)(json, 'code') ? undefined : json['code'],
        'organization': !(0, runtime_js_1.exists)(json, 'organization') ? undefined : (0, CreateOrganizationResponseOrganization_js_1.CreateOrganizationResponseOrganizationFromJSON)(json['organization']),
    };
}
exports.CreateOrganizationResponseFromJSONTyped = CreateOrganizationResponseFromJSONTyped;
function CreateOrganizationResponseToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'message': value.message,
        'code': value.code,
        'organization': (0, CreateOrganizationResponseOrganization_js_1.CreateOrganizationResponseOrganizationToJSON)(value.organization),
    };
}
exports.CreateOrganizationResponseToJSON = CreateOrganizationResponseToJSON;
