// @kinde-oss/kinde-typescript-sdk/dist-cjs/models/CreateUserResponse.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateUserResponseToJSON = exports.CreateUserResponseFromJSONTyped = exports.CreateUserResponseFromJSON = exports.instanceOfCreateUserResponse = void 0;
var runtime_js_1 = require("../runtime.js");
var UserIdentity_js_1 = require("./UserIdentity.js");
/**
 * Check if a given object implements the CreateUserResponse interface.
 */
function instanceOfCreateUserResponse(value) {
    var isInstance = true;
    return isInstance;
}
exports.instanceOfCreateUserResponse = instanceOfCreateUserResponse;
function CreateUserResponseFromJSON(json) {
    return CreateUserResponseFromJSONTyped(json, false);
}
exports.CreateUserResponseFromJSON = CreateUserResponseFromJSON;
function CreateUserResponseFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'id': !(0, runtime_js_1.exists)(json, 'id') ? undefined : json['id'],
        'created': !(0, runtime_js_1.exists)(json, 'created') ? undefined : json['created'],
        'identities': !(0, runtime_js_1.exists)(json, 'identities') ? undefined : (json['identities'].map(UserIdentity_js_1.UserIdentityFromJSON)),
    };
}
exports.CreateUserResponseFromJSONTyped = CreateUserResponseFromJSONTyped;
function CreateUserResponseToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'id': value.id,
        'created': value.created,
        'identities': value.identities === undefined ? undefined : (value.identities.map(UserIdentity_js_1.UserIdentityToJSON)),
    };
}
exports.CreateUserResponseToJSON = CreateUserResponseToJSON;
