// @kinde-oss/kinde-typescript-sdk/dist-cjs/models/CreateSubscriberSuccessResponseSubscriber.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateSubscriberSuccessResponseSubscriberToJSON = exports.CreateSubscriberSuccessResponseSubscriberFromJSONTyped = exports.CreateSubscriberSuccessResponseSubscriberFromJSON = exports.instanceOfCreateSubscriberSuccessResponseSubscriber = void 0;
var runtime_js_1 = require("../runtime.js");
/**
 * Check if a given object implements the CreateSubscriberSuccessResponseSubscriber interface.
 */
function instanceOfCreateSubscriberSuccessResponseSubscriber(value) {
    var isInstance = true;
    return isInstance;
}
exports.instanceOfCreateSubscriberSuccessResponseSubscriber = instanceOfCreateSubscriberSuccessResponseSubscriber;
function CreateSubscriberSuccessResponseSubscriberFromJSON(json) {
    return CreateSubscriberSuccessResponseSubscriberFromJSONTyped(json, false);
}
exports.CreateSubscriberSuccessResponseSubscriberFromJSON = CreateSubscriberSuccessResponseSubscriberFromJSON;
function CreateSubscriberSuccessResponseSubscriberFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'subscriberId': !(0, runtime_js_1.exists)(json, 'subscriber_id') ? undefined : json['subscriber_id'],
    };
}
exports.CreateSubscriberSuccessResponseSubscriberFromJSONTyped = CreateSubscriberSuccessResponseSubscriberFromJSONTyped;
function CreateSubscriberSuccessResponseSubscriberToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'subscriber_id': value.subscriberId,
    };
}
exports.CreateSubscriberSuccessResponseSubscriberToJSON = CreateSubscriberSuccessResponseSubscriberToJSON;
