// @kinde-oss/kinde-typescript-sdk/dist-cjs/models/CreateUserRequestIdentitiesInner.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateUserRequestIdentitiesInnerToJSON = exports.CreateUserRequestIdentitiesInnerFromJSONTyped = exports.CreateUserRequestIdentitiesInnerFromJSON = exports.instanceOfCreateUserRequestIdentitiesInner = exports.CreateUserRequestIdentitiesInnerTypeEnum = void 0;
var runtime_js_1 = require("../runtime.js");
var CreateUserRequestIdentitiesInnerDetails_js_1 = require("./CreateUserRequestIdentitiesInnerDetails.js");
/**
 * @export
 */
exports.CreateUserRequestIdentitiesInnerTypeEnum = {
    Email: 'email'
};
/**
 * Check if a given object implements the CreateUserRequestIdentitiesInner interface.
 */
function instanceOfCreateUserRequestIdentitiesInner(value) {
    var isInstance = true;
    return isInstance;
}
exports.instanceOfCreateUserRequestIdentitiesInner = instanceOfCreateUserRequestIdentitiesInner;
function CreateUserRequestIdentitiesInnerFromJSON(json) {
    return CreateUserRequestIdentitiesInnerFromJSONTyped(json, false);
}
exports.CreateUserRequestIdentitiesInnerFromJSON = CreateUserRequestIdentitiesInnerFromJSON;
function CreateUserRequestIdentitiesInnerFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'type': !(0, runtime_js_1.exists)(json, 'type') ? undefined : json['type'],
        'details': !(0, runtime_js_1.exists)(json, 'details') ? undefined : (0, CreateUserRequestIdentitiesInnerDetails_js_1.CreateUserRequestIdentitiesInnerDetailsFromJSON)(json['details']),
    };
}
exports.CreateUserRequestIdentitiesInnerFromJSONTyped = CreateUserRequestIdentitiesInnerFromJSONTyped;
function CreateUserRequestIdentitiesInnerToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'type': value.type,
        'details': (0, CreateUserRequestIdentitiesInnerDetails_js_1.CreateUserRequestIdentitiesInnerDetailsToJSON)(value.details),
    };
}
exports.CreateUserRequestIdentitiesInnerToJSON = CreateUserRequestIdentitiesInnerToJSON;
