// @kinde-oss/kinde-typescript-sdk/dist-cjs/models/CreateApplicationRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateApplicationRequestToJSON = exports.CreateApplicationRequestFromJSONTyped = exports.CreateApplicationRequestFromJSON = exports.instanceOfCreateApplicationRequest = exports.CreateApplicationRequestTypeEnum = void 0;
var runtime_js_1 = require("../runtime.js");
/**
 * @export
 */
exports.CreateApplicationRequestTypeEnum = {
    Reg: 'reg',
    Spa: 'spa',
    M2m: 'm2m'
};
/**
 * Check if a given object implements the CreateApplicationRequest interface.
 */
function instanceOfCreateApplicationRequest(value) {
    var isInstance = true;
    return isInstance;
}
exports.instanceOfCreateApplicationRequest = instanceOfCreateApplicationRequest;
function CreateApplicationRequestFromJSON(json) {
    return CreateApplicationRequestFromJSONTyped(json, false);
}
exports.CreateApplicationRequestFromJSON = CreateApplicationRequestFromJSON;
function CreateApplicationRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': !(0, runtime_js_1.exists)(json, 'name') ? undefined : json['name'],
        'type': !(0, runtime_js_1.exists)(json, 'type') ? undefined : json['type'],
    };
}
exports.CreateApplicationRequestFromJSONTyped = CreateApplicationRequestFromJSONTyped;
function CreateApplicationRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'type': value.type,
    };
}
exports.CreateApplicationRequestToJSON = CreateApplicationRequestToJSON;
