// @kinde-oss/kinde-typescript-sdk/dist-cjs/models/CreateRoleRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateRoleRequestToJSON = exports.CreateRoleRequestFromJSONTyped = exports.CreateRoleRequestFromJSON = exports.instanceOfCreateRoleRequest = void 0;
var runtime_js_1 = require("../runtime.js");
/**
 * Check if a given object implements the CreateRoleRequest interface.
 */
function instanceOfCreateRoleRequest(value) {
    var isInstance = true;
    return isInstance;
}
exports.instanceOfCreateRoleRequest = instanceOfCreateRoleRequest;
function CreateRoleRequestFromJSON(json) {
    return CreateRoleRequestFromJSONTyped(json, false);
}
exports.CreateRoleRequestFromJSON = CreateRoleRequestFromJSON;
function CreateRoleRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': !(0, runtime_js_1.exists)(json, 'name') ? undefined : json['name'],
        'description': !(0, runtime_js_1.exists)(json, 'description') ? undefined : json['description'],
        'key': !(0, runtime_js_1.exists)(json, 'key') ? undefined : json['key'],
        'isDefaultRole': !(0, runtime_js_1.exists)(json, 'is_default_role') ? undefined : json['is_default_role'],
    };
}
exports.CreateRoleRequestFromJSONTyped = CreateRoleRequestFromJSONTyped;
function CreateRoleRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'description': value.description,
        'key': value.key,
        'is_default_role': value.isDefaultRole,
    };
}
exports.CreateRoleRequestToJSON = CreateRoleRequestToJSON;
