// @kinde-oss/kinde-typescript-sdk/dist-cjs/models/CreateOrganizationRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateOrganizationRequestToJSON = exports.CreateOrganizationRequestFromJSONTyped = exports.CreateOrganizationRequestFromJSON = exports.instanceOfCreateOrganizationRequest = exports.CreateOrganizationRequestFeatureFlagsEnum = void 0;
var runtime_js_1 = require("../runtime.js");
/**
 * @export
 */
exports.CreateOrganizationRequestFeatureFlagsEnum = {
    Str: 'str',
    Int: 'int',
    Bool: 'bool'
};
/**
 * Check if a given object implements the CreateOrganizationRequest interface.
 */
function instanceOfCreateOrganizationRequest(value) {
    var isInstance = true;
    isInstance = isInstance && "name" in value;
    return isInstance;
}
exports.instanceOfCreateOrganizationRequest = instanceOfCreateOrganizationRequest;
function CreateOrganizationRequestFromJSON(json) {
    return CreateOrganizationRequestFromJSONTyped(json, false);
}
exports.CreateOrganizationRequestFromJSON = CreateOrganizationRequestFromJSON;
function CreateOrganizationRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': json['name'],
        'featureFlags': !(0, runtime_js_1.exists)(json, 'feature_flags') ? undefined : json['feature_flags'],
        'externalId': !(0, runtime_js_1.exists)(json, 'external_id') ? undefined : json['external_id'],
        'backgroundColor': !(0, runtime_js_1.exists)(json, 'background_color') ? undefined : json['background_color'],
        'buttonColor': !(0, runtime_js_1.exists)(json, 'button_color') ? undefined : json['button_color'],
        'buttonTextColor': !(0, runtime_js_1.exists)(json, 'button_text_color') ? undefined : json['button_text_color'],
        'linkColor': !(0, runtime_js_1.exists)(json, 'link_color') ? undefined : json['link_color'],
        'backgroundColorDark': !(0, runtime_js_1.exists)(json, 'background_color_dark') ? undefined : json['background_color_dark'],
        'buttonColorDark': !(0, runtime_js_1.exists)(json, 'button_color_dark') ? undefined : json['button_color_dark'],
        'buttonTextColorDark': !(0, runtime_js_1.exists)(json, 'button_text_color_dark') ? undefined : json['button_text_color_dark'],
        'linkColorDark': !(0, runtime_js_1.exists)(json, 'link_color_dark') ? undefined : json['link_color_dark'],
        'themeCode': !(0, runtime_js_1.exists)(json, 'theme_code') ? undefined : json['theme_code'],
        'handle': !(0, runtime_js_1.exists)(json, 'handle') ? undefined : json['handle'],
        'isAllowRegistrations': !(0, runtime_js_1.exists)(json, 'is_allow_registrations') ? undefined : json['is_allow_registrations'],
    };
}
exports.CreateOrganizationRequestFromJSONTyped = CreateOrganizationRequestFromJSONTyped;
function CreateOrganizationRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'feature_flags': value.featureFlags,
        'external_id': value.externalId,
        'background_color': value.backgroundColor,
        'button_color': value.buttonColor,
        'button_text_color': value.buttonTextColor,
        'link_color': value.linkColor,
        'background_color_dark': value.backgroundColorDark,
        'button_color_dark': value.buttonColorDark,
        'button_text_color_dark': value.buttonTextColorDark,
        'link_color_dark': value.linkColorDark,
        'theme_code': value.themeCode,
        'handle': value.handle,
        'is_allow_registrations': value.isAllowRegistrations,
    };
}
exports.CreateOrganizationRequestToJSON = CreateOrganizationRequestToJSON;
