// @kinde-oss/kinde-typescript-sdk/dist-cjs/models/CreateOrganizationUserRoleRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateOrganizationUserRoleRequestToJSON = exports.CreateOrganizationUserRoleRequestFromJSONTyped = exports.CreateOrganizationUserRoleRequestFromJSON = exports.instanceOfCreateOrganizationUserRoleRequest = void 0;
var runtime_js_1 = require("../runtime.js");
/**
 * Check if a given object implements the CreateOrganizationUserRoleRequest interface.
 */
function instanceOfCreateOrganizationUserRoleRequest(value) {
    var isInstance = true;
    return isInstance;
}
exports.instanceOfCreateOrganizationUserRoleRequest = instanceOfCreateOrganizationUserRoleRequest;
function CreateOrganizationUserRoleRequestFromJSON(json) {
    return CreateOrganizationUserRoleRequestFromJSONTyped(json, false);
}
exports.CreateOrganizationUserRoleRequestFromJSON = CreateOrganizationUserRoleRequestFromJSON;
function CreateOrganizationUserRoleRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'roleId': !(0, runtime_js_1.exists)(json, 'role_id') ? undefined : json['role_id'],
    };
}
exports.CreateOrganizationUserRoleRequestFromJSONTyped = CreateOrganizationUserRoleRequestFromJSONTyped;
function CreateOrganizationUserRoleRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'role_id': value.roleId,
    };
}
exports.CreateOrganizationUserRoleRequestToJSON = CreateOrganizationUserRoleRequestToJSON;
