// @kinde-oss/kinde-typescript-sdk/dist-cjs/models/CreateFeatureFlagRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateFeatureFlagRequestToJSON = exports.CreateFeatureFlagRequestFromJSONTyped = exports.CreateFeatureFlagRequestFromJSON = exports.instanceOfCreateFeatureFlagRequest = exports.CreateFeatureFlagRequestAllowOverrideLevelEnum = exports.CreateFeatureFlagRequestTypeEnum = void 0;
var runtime_js_1 = require("../runtime.js");
/**
 * @export
 */
exports.CreateFeatureFlagRequestTypeEnum = {
    Str: 'str',
    Int: 'int',
    Bool: 'bool'
};
/**
 * @export
 */
exports.CreateFeatureFlagRequestAllowOverrideLevelEnum = {
    Env: 'env',
    Org: 'org',
    Usr: 'usr'
};
/**
 * Check if a given object implements the CreateFeatureFlagRequest interface.
 */
function instanceOfCreateFeatureFlagRequest(value) {
    var isInstance = true;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "key" in value;
    isInstance = isInstance && "type" in value;
    isInstance = isInstance && "defaultValue" in value;
    return isInstance;
}
exports.instanceOfCreateFeatureFlagRequest = instanceOfCreateFeatureFlagRequest;
function CreateFeatureFlagRequestFromJSON(json) {
    return CreateFeatureFlagRequestFromJSONTyped(json, false);
}
exports.CreateFeatureFlagRequestFromJSON = CreateFeatureFlagRequestFromJSON;
function CreateFeatureFlagRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': json['name'],
        'description': !(0, runtime_js_1.exists)(json, 'description') ? undefined : json['description'],
        'key': json['key'],
        'type': json['type'],
        'allowOverrideLevel': !(0, runtime_js_1.exists)(json, 'allow_override_level') ? undefined : json['allow_override_level'],
        'defaultValue': json['default_value'],
    };
}
exports.CreateFeatureFlagRequestFromJSONTyped = CreateFeatureFlagRequestFromJSONTyped;
function CreateFeatureFlagRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'description': value.description,
        'key': value.key,
        'type': value.type,
        'allow_override_level': value.allowOverrideLevel,
        'default_value': value.defaultValue,
    };
}
exports.CreateFeatureFlagRequestToJSON = CreateFeatureFlagRequestToJSON;
