// @kinde-oss/kinde-typescript-sdk/dist-cjs/models/CreateOrganizationResponseOrganization.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateOrganizationResponseOrganizationToJSON = exports.CreateOrganizationResponseOrganizationFromJSONTyped = exports.CreateOrganizationResponseOrganizationFromJSON = exports.instanceOfCreateOrganizationResponseOrganization = void 0;
var runtime_js_1 = require("../runtime.js");
/**
 * Check if a given object implements the CreateOrganizationResponseOrganization interface.
 */
function instanceOfCreateOrganizationResponseOrganization(value) {
    var isInstance = true;
    return isInstance;
}
exports.instanceOfCreateOrganizationResponseOrganization = instanceOfCreateOrganizationResponseOrganization;
function CreateOrganizationResponseOrganizationFromJSON(json) {
    return CreateOrganizationResponseOrganizationFromJSONTyped(json, false);
}
exports.CreateOrganizationResponseOrganizationFromJSON = CreateOrganizationResponseOrganizationFromJSON;
function CreateOrganizationResponseOrganizationFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'code': !(0, runtime_js_1.exists)(json, 'code') ? undefined : json['code'],
    };
}
exports.CreateOrganizationResponseOrganizationFromJSONTyped = CreateOrganizationResponseOrganizationFromJSONTyped;
function CreateOrganizationResponseOrganizationToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'code': value.code,
    };
}
exports.CreateOrganizationResponseOrganizationToJSON = CreateOrganizationResponseOrganizationToJSON;
