// @kinde-oss/kinde-typescript-sdk/dist-cjs/models/CreateUserRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateUserRequestToJSON = exports.CreateUserRequestFromJSONTyped = exports.CreateUserRequestFromJSON = exports.instanceOfCreateUserRequest = void 0;
var runtime_js_1 = require("../runtime.js");
var CreateUserRequestIdentitiesInner_js_1 = require("./CreateUserRequestIdentitiesInner.js");
var CreateUserRequestProfile_js_1 = require("./CreateUserRequestProfile.js");
/**
 * Check if a given object implements the CreateUserRequest interface.
 */
function instanceOfCreateUserRequest(value) {
    var isInstance = true;
    return isInstance;
}
exports.instanceOfCreateUserRequest = instanceOfCreateUserRequest;
function CreateUserRequestFromJSON(json) {
    return CreateUserRequestFromJSONTyped(json, false);
}
exports.CreateUserRequestFromJSON = CreateUserRequestFromJSON;
function CreateUserRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'profile': !(0, runtime_js_1.exists)(json, 'profile') ? undefined : (0, CreateUserRequestProfile_js_1.CreateUserRequestProfileFromJSON)(json['profile']),
        'organizationCode': !(0, runtime_js_1.exists)(json, 'organization_code') ? undefined : json['organization_code'],
        'identities': !(0, runtime_js_1.exists)(json, 'identities') ? undefined : (json['identities'].map(CreateUserRequestIdentitiesInner_js_1.CreateUserRequestIdentitiesInnerFromJSON)),
    };
}
exports.CreateUserRequestFromJSONTyped = CreateUserRequestFromJSONTyped;
function CreateUserRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'profile': (0, CreateUserRequestProfile_js_1.CreateUserRequestProfileToJSON)(value.profile),
        'organization_code': value.organizationCode,
        'identities': value.identities === undefined ? undefined : (value.identities.map(CreateUserRequestIdentitiesInner_js_1.CreateUserRequestIdentitiesInnerToJSON)),
    };
}
exports.CreateUserRequestToJSON = CreateUserRequestToJSON;
