// @kinde-oss/kinde-typescript-sdk/dist-cjs/models/CreateApplicationResponseApplication.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateApplicationResponseApplicationToJSON = exports.CreateApplicationResponseApplicationFromJSONTyped = exports.CreateApplicationResponseApplicationFromJSON = exports.instanceOfCreateApplicationResponseApplication = void 0;
var runtime_js_1 = require("../runtime.js");
/**
 * Check if a given object implements the CreateApplicationResponseApplication interface.
 */
function instanceOfCreateApplicationResponseApplication(value) {
    var isInstance = true;
    return isInstance;
}
exports.instanceOfCreateApplicationResponseApplication = instanceOfCreateApplicationResponseApplication;
function CreateApplicationResponseApplicationFromJSON(json) {
    return CreateApplicationResponseApplicationFromJSONTyped(json, false);
}
exports.CreateApplicationResponseApplicationFromJSON = CreateApplicationResponseApplicationFromJSON;
function CreateApplicationResponseApplicationFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'id': !(0, runtime_js_1.exists)(json, 'id') ? undefined : json['id'],
        'clientId': !(0, runtime_js_1.exists)(json, 'client_id') ? undefined : json['client_id'],
        'clientSecret': !(0, runtime_js_1.exists)(json, 'client_secret') ? undefined : json['client_secret'],
    };
}
exports.CreateApplicationResponseApplicationFromJSONTyped = CreateApplicationResponseApplicationFromJSONTyped;
function CreateApplicationResponseApplicationToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'id': value.id,
        'client_id': value.clientId,
        'client_secret': value.clientSecret,
    };
}
exports.CreateApplicationResponseApplicationToJSON = CreateApplicationResponseApplicationToJSON;
