// @kinde-oss/kinde-typescript-sdk/dist-cjs/models/CreateApplicationResponse.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateApplicationResponseToJSON = exports.CreateApplicationResponseFromJSONTyped = exports.CreateApplicationResponseFromJSON = exports.instanceOfCreateApplicationResponse = void 0;
var runtime_js_1 = require("../runtime.js");
var CreateApplicationResponseApplication_js_1 = require("./CreateApplicationResponseApplication.js");
/**
 * Check if a given object implements the CreateApplicationResponse interface.
 */
function instanceOfCreateApplicationResponse(value) {
    var isInstance = true;
    return isInstance;
}
exports.instanceOfCreateApplicationResponse = instanceOfCreateApplicationResponse;
function CreateApplicationResponseFromJSON(json) {
    return CreateApplicationResponseFromJSONTyped(json, false);
}
exports.CreateApplicationResponseFromJSON = CreateApplicationResponseFromJSON;
function CreateApplicationResponseFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'code': !(0, runtime_js_1.exists)(json, 'code') ? undefined : json['code'],
        'message': !(0, runtime_js_1.exists)(json, 'message') ? undefined : json['message'],
        'application': !(0, runtime_js_1.exists)(json, 'application') ? undefined : (0, CreateApplicationResponseApplication_js_1.CreateApplicationResponseApplicationFromJSON)(json['application']),
    };
}
exports.CreateApplicationResponseFromJSONTyped = CreateApplicationResponseFromJSONTyped;
function CreateApplicationResponseToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'code': value.code,
        'message': value.message,
        'application': (0, CreateApplicationResponseApplication_js_1.CreateApplicationResponseApplicationToJSON)(value.application),
    };
}
exports.CreateApplicationResponseToJSON = CreateApplicationResponseToJSON;
