// @kinde-oss/kinde-typescript-sdk/dist-cjs/models/CreateOrganizationUserPermissionRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateOrganizationUserPermissionRequestToJSON = exports.CreateOrganizationUserPermissionRequestFromJSONTyped = exports.CreateOrganizationUserPermissionRequestFromJSON = exports.instanceOfCreateOrganizationUserPermissionRequest = void 0;
var runtime_js_1 = require("../runtime.js");
/**
 * Check if a given object implements the CreateOrganizationUserPermissionRequest interface.
 */
function instanceOfCreateOrganizationUserPermissionRequest(value) {
    var isInstance = true;
    return isInstance;
}
exports.instanceOfCreateOrganizationUserPermissionRequest = instanceOfCreateOrganizationUserPermissionRequest;
function CreateOrganizationUserPermissionRequestFromJSON(json) {
    return CreateOrganizationUserPermissionRequestFromJSONTyped(json, false);
}
exports.CreateOrganizationUserPermissionRequestFromJSON = CreateOrganizationUserPermissionRequestFromJSON;
function CreateOrganizationUserPermissionRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'permissionId': !(0, runtime_js_1.exists)(json, 'permission_id') ? undefined : json['permission_id'],
    };
}
exports.CreateOrganizationUserPermissionRequestFromJSONTyped = CreateOrganizationUserPermissionRequestFromJSONTyped;
function CreateOrganizationUserPermissionRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'permission_id': value.permissionId,
    };
}
exports.CreateOrganizationUserPermissionRequestToJSON = CreateOrganizationUserPermissionRequestToJSON;
