// @kinde-oss/kinde-typescript-sdk/dist-cjs/models/CreateUserRequestIdentitiesInnerDetails.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateUserRequestIdentitiesInnerDetailsToJSON = exports.CreateUserRequestIdentitiesInnerDetailsFromJSONTyped = exports.CreateUserRequestIdentitiesInnerDetailsFromJSON = exports.instanceOfCreateUserRequestIdentitiesInnerDetails = void 0;
var runtime_js_1 = require("../runtime.js");
/**
 * Check if a given object implements the CreateUserRequestIdentitiesInnerDetails interface.
 */
function instanceOfCreateUserRequestIdentitiesInnerDetails(value) {
    var isInstance = true;
    return isInstance;
}
exports.instanceOfCreateUserRequestIdentitiesInnerDetails = instanceOfCreateUserRequestIdentitiesInnerDetails;
function CreateUserRequestIdentitiesInnerDetailsFromJSON(json) {
    return CreateUserRequestIdentitiesInnerDetailsFromJSONTyped(json, false);
}
exports.CreateUserRequestIdentitiesInnerDetailsFromJSON = CreateUserRequestIdentitiesInnerDetailsFromJSON;
function CreateUserRequestIdentitiesInnerDetailsFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'email': !(0, runtime_js_1.exists)(json, 'email') ? undefined : json['email'],
    };
}
exports.CreateUserRequestIdentitiesInnerDetailsFromJSONTyped = CreateUserRequestIdentitiesInnerDetailsFromJSONTyped;
function CreateUserRequestIdentitiesInnerDetailsToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'email': value.email,
    };
}
exports.CreateUserRequestIdentitiesInnerDetailsToJSON = CreateUserRequestIdentitiesInnerDetailsToJSON;
