// @kinde-oss/kinde-typescript-sdk/dist-cjs/models/GetApplicationsResponse.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetApplicationsResponseToJSON = exports.GetApplicationsResponseFromJSONTyped = exports.GetApplicationsResponseFromJSON = exports.instanceOfGetApplicationsResponse = void 0;
var runtime_js_1 = require("../runtime.js");
var Applications_js_1 = require("./Applications.js");
/**
 * Check if a given object implements the GetApplicationsResponse interface.
 */
function instanceOfGetApplicationsResponse(value) {
    var isInstance = true;
    return isInstance;
}
exports.instanceOfGetApplicationsResponse = instanceOfGetApplicationsResponse;
function GetApplicationsResponseFromJSON(json) {
    return GetApplicationsResponseFromJSONTyped(json, false);
}
exports.GetApplicationsResponseFromJSON = GetApplicationsResponseFromJSON;
function GetApplicationsResponseFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'code': !(0, runtime_js_1.exists)(json, 'code') ? undefined : json['code'],
        'message': !(0, runtime_js_1.exists)(json, 'message') ? undefined : json['message'],
        'applications': !(0, runtime_js_1.exists)(json, 'applications') ? undefined : (json['applications'].map(Applications_js_1.ApplicationsFromJSON)),
        'nextToken': !(0, runtime_js_1.exists)(json, 'next_token') ? undefined : json['next_token'],
    };
}
exports.GetApplicationsResponseFromJSONTyped = GetApplicationsResponseFromJSONTyped;
function GetApplicationsResponseToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'code': value.code,
        'message': value.message,
        'applications': value.applications === undefined ? undefined : (value.applications.map(Applications_js_1.ApplicationsToJSON)),
        'next_token': value.nextToken,
    };
}
exports.GetApplicationsResponseToJSON = GetApplicationsResponseToJSON;
