// @kinde-oss/kinde-typescript-sdk/dist-cjs/models/GetApplicationResponseApplication.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetApplicationResponseApplicationToJSON = exports.GetApplicationResponseApplicationFromJSONTyped = exports.GetApplicationResponseApplicationFromJSON = exports.instanceOfGetApplicationResponseApplication = void 0;
var runtime_js_1 = require("../runtime.js");
/**
 * Check if a given object implements the GetApplicationResponseApplication interface.
 */
function instanceOfGetApplicationResponseApplication(value) {
    var isInstance = true;
    return isInstance;
}
exports.instanceOfGetApplicationResponseApplication = instanceOfGetApplicationResponseApplication;
function GetApplicationResponseApplicationFromJSON(json) {
    return GetApplicationResponseApplicationFromJSONTyped(json, false);
}
exports.GetApplicationResponseApplicationFromJSON = GetApplicationResponseApplicationFromJSON;
function GetApplicationResponseApplicationFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'id': !(0, runtime_js_1.exists)(json, 'id') ? undefined : json['id'],
        'name': !(0, runtime_js_1.exists)(json, 'name') ? undefined : json['name'],
        'type': !(0, runtime_js_1.exists)(json, 'type') ? undefined : json['type'],
        'clientId': !(0, runtime_js_1.exists)(json, 'client_id') ? undefined : json['client_id'],
        'clientSecret': !(0, runtime_js_1.exists)(json, 'client_secret') ? undefined : json['client_secret'],
    };
}
exports.GetApplicationResponseApplicationFromJSONTyped = GetApplicationResponseApplicationFromJSONTyped;
function GetApplicationResponseApplicationToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'id': value.id,
        'name': value.name,
        'type': value.type,
        'client_id': value.clientId,
        'client_secret': value.clientSecret,
    };
}
exports.GetApplicationResponseApplicationToJSON = GetApplicationResponseApplicationToJSON;
