// @kinde-oss/kinde-typescript-sdk/dist-cjs/models/GetApplicationResponse.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetApplicationResponseToJSON = exports.GetApplicationResponseFromJSONTyped = exports.GetApplicationResponseFromJSON = exports.instanceOfGetApplicationResponse = void 0;
var runtime_js_1 = require("../runtime.js");
var GetApplicationResponseApplication_js_1 = require("./GetApplicationResponseApplication.js");
/**
 * Check if a given object implements the GetApplicationResponse interface.
 */
function instanceOfGetApplicationResponse(value) {
    var isInstance = true;
    return isInstance;
}
exports.instanceOfGetApplicationResponse = instanceOfGetApplicationResponse;
function GetApplicationResponseFromJSON(json) {
    return GetApplicationResponseFromJSONTyped(json, false);
}
exports.GetApplicationResponseFromJSON = GetApplicationResponseFromJSON;
function GetApplicationResponseFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'code': !(0, runtime_js_1.exists)(json, 'code') ? undefined : json['code'],
        'message': !(0, runtime_js_1.exists)(json, 'message') ? undefined : json['message'],
        'application': !(0, runtime_js_1.exists)(json, 'application') ? undefined : (0, GetApplicationResponseApplication_js_1.GetApplicationResponseApplicationFromJSON)(json['application']),
    };
}
exports.GetApplicationResponseFromJSONTyped = GetApplicationResponseFromJSONTyped;
function GetApplicationResponseToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'code': value.code,
        'message': value.message,
        'application': (0, GetApplicationResponseApplication_js_1.GetApplicationResponseApplicationToJSON)(value.application),
    };
}
exports.GetApplicationResponseToJSON = GetApplicationResponseToJSON;
