// @kinde-oss/kinde-typescript-sdk/dist-cjs/models/GetOrganizationUsersResponse.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetOrganizationUsersResponseToJSON = exports.GetOrganizationUsersResponseFromJSONTyped = exports.GetOrganizationUsersResponseFromJSON = exports.instanceOfGetOrganizationUsersResponse = void 0;
var runtime_js_1 = require("../runtime.js");
var OrganizationUser_js_1 = require("./OrganizationUser.js");
/**
 * Check if a given object implements the GetOrganizationUsersResponse interface.
 */
function instanceOfGetOrganizationUsersResponse(value) {
    var isInstance = true;
    return isInstance;
}
exports.instanceOfGetOrganizationUsersResponse = instanceOfGetOrganizationUsersResponse;
function GetOrganizationUsersResponseFromJSON(json) {
    return GetOrganizationUsersResponseFromJSONTyped(json, false);
}
exports.GetOrganizationUsersResponseFromJSON = GetOrganizationUsersResponseFromJSON;
function GetOrganizationUsersResponseFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'code': !(0, runtime_js_1.exists)(json, 'code') ? undefined : json['code'],
        'message': !(0, runtime_js_1.exists)(json, 'message') ? undefined : json['message'],
        'organizationUsers': !(0, runtime_js_1.exists)(json, 'organization_users') ? undefined : (json['organization_users'].map(OrganizationUser_js_1.OrganizationUserFromJSON)),
        'nextToken': !(0, runtime_js_1.exists)(json, 'next_token') ? undefined : json['next_token'],
    };
}
exports.GetOrganizationUsersResponseFromJSONTyped = GetOrganizationUsersResponseFromJSONTyped;
function GetOrganizationUsersResponseToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'code': value.code,
        'message': value.message,
        'organization_users': value.organizationUsers === undefined ? undefined : (value.organizationUsers.map(OrganizationUser_js_1.OrganizationUserToJSON)),
        'next_token': value.nextToken,
    };
}
exports.GetOrganizationUsersResponseToJSON = GetOrganizationUsersResponseToJSON;
