// @kinde-oss/kinde-typescript-sdk/dist-cjs/models/GetOrganizationFeatureFlagsResponse.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetOrganizationFeatureFlagsResponseToJSON = exports.GetOrganizationFeatureFlagsResponseFromJSONTyped = exports.GetOrganizationFeatureFlagsResponseFromJSON = exports.instanceOfGetOrganizationFeatureFlagsResponse = void 0;
var runtime_js_1 = require("../runtime.js");
var GetOrganizationFeatureFlagsResponseFeatureFlagsValue_js_1 = require("./GetOrganizationFeatureFlagsResponseFeatureFlagsValue.js");
/**
 * Check if a given object implements the GetOrganizationFeatureFlagsResponse interface.
 */
function instanceOfGetOrganizationFeatureFlagsResponse(value) {
    var isInstance = true;
    return isInstance;
}
exports.instanceOfGetOrganizationFeatureFlagsResponse = instanceOfGetOrganizationFeatureFlagsResponse;
function GetOrganizationFeatureFlagsResponseFromJSON(json) {
    return GetOrganizationFeatureFlagsResponseFromJSONTyped(json, false);
}
exports.GetOrganizationFeatureFlagsResponseFromJSON = GetOrganizationFeatureFlagsResponseFromJSON;
function GetOrganizationFeatureFlagsResponseFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'code': !(0, runtime_js_1.exists)(json, 'code') ? undefined : json['code'],
        'message': !(0, runtime_js_1.exists)(json, 'message') ? undefined : json['message'],
        'featureFlags': !(0, runtime_js_1.exists)(json, 'feature_flags') ? undefined : ((0, runtime_js_1.mapValues)(json['feature_flags'], GetOrganizationFeatureFlagsResponseFeatureFlagsValue_js_1.GetOrganizationFeatureFlagsResponseFeatureFlagsValueFromJSON)),
    };
}
exports.GetOrganizationFeatureFlagsResponseFromJSONTyped = GetOrganizationFeatureFlagsResponseFromJSONTyped;
function GetOrganizationFeatureFlagsResponseToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'code': value.code,
        'message': value.message,
        'feature_flags': value.featureFlags === undefined ? undefined : ((0, runtime_js_1.mapValues)(value.featureFlags, GetOrganizationFeatureFlagsResponseFeatureFlagsValue_js_1.GetOrganizationFeatureFlagsResponseFeatureFlagsValueToJSON)),
    };
}
exports.GetOrganizationFeatureFlagsResponseToJSON = GetOrganizationFeatureFlagsResponseToJSON;
