// @kinde-oss/kinde-typescript-sdk/dist-cjs/models/GetOrganizationFeatureFlagsResponseFeatureFlagsValue.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetOrganizationFeatureFlagsResponseFeatureFlagsValueToJSON = exports.GetOrganizationFeatureFlagsResponseFeatureFlagsValueFromJSONTyped = exports.GetOrganizationFeatureFlagsResponseFeatureFlagsValueFromJSON = exports.instanceOfGetOrganizationFeatureFlagsResponseFeatureFlagsValue = exports.GetOrganizationFeatureFlagsResponseFeatureFlagsValueTypeEnum = void 0;
var runtime_js_1 = require("../runtime.js");
/**
 * @export
 */
exports.GetOrganizationFeatureFlagsResponseFeatureFlagsValueTypeEnum = {
    Str: 'str',
    Int: 'int',
    Bool: 'bool'
};
/**
 * Check if a given object implements the GetOrganizationFeatureFlagsResponseFeatureFlagsValue interface.
 */
function instanceOfGetOrganizationFeatureFlagsResponseFeatureFlagsValue(value) {
    var isInstance = true;
    return isInstance;
}
exports.instanceOfGetOrganizationFeatureFlagsResponseFeatureFlagsValue = instanceOfGetOrganizationFeatureFlagsResponseFeatureFlagsValue;
function GetOrganizationFeatureFlagsResponseFeatureFlagsValueFromJSON(json) {
    return GetOrganizationFeatureFlagsResponseFeatureFlagsValueFromJSONTyped(json, false);
}
exports.GetOrganizationFeatureFlagsResponseFeatureFlagsValueFromJSON = GetOrganizationFeatureFlagsResponseFeatureFlagsValueFromJSON;
function GetOrganizationFeatureFlagsResponseFeatureFlagsValueFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'type': !(0, runtime_js_1.exists)(json, 'type') ? undefined : json['type'],
        'value': !(0, runtime_js_1.exists)(json, 'value') ? undefined : json['value'],
    };
}
exports.GetOrganizationFeatureFlagsResponseFeatureFlagsValueFromJSONTyped = GetOrganizationFeatureFlagsResponseFeatureFlagsValueFromJSONTyped;
function GetOrganizationFeatureFlagsResponseFeatureFlagsValueToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'type': value.type,
        'value': value.value,
    };
}
exports.GetOrganizationFeatureFlagsResponseFeatureFlagsValueToJSON = GetOrganizationFeatureFlagsResponseFeatureFlagsValueToJSON;
