// @kinde-oss/kinde-typescript-sdk/dist-cjs/models/GetOrganizationsUserPermissionsResponse.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetOrganizationsUserPermissionsResponseToJSON = exports.GetOrganizationsUserPermissionsResponseFromJSONTyped = exports.GetOrganizationsUserPermissionsResponseFromJSON = exports.instanceOfGetOrganizationsUserPermissionsResponse = void 0;
var runtime_js_1 = require("../runtime.js");
var OrganizationUserPermission_js_1 = require("./OrganizationUserPermission.js");
/**
 * Check if a given object implements the GetOrganizationsUserPermissionsResponse interface.
 */
function instanceOfGetOrganizationsUserPermissionsResponse(value) {
    var isInstance = true;
    return isInstance;
}
exports.instanceOfGetOrganizationsUserPermissionsResponse = instanceOfGetOrganizationsUserPermissionsResponse;
function GetOrganizationsUserPermissionsResponseFromJSON(json) {
    return GetOrganizationsUserPermissionsResponseFromJSONTyped(json, false);
}
exports.GetOrganizationsUserPermissionsResponseFromJSON = GetOrganizationsUserPermissionsResponseFromJSON;
function GetOrganizationsUserPermissionsResponseFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'code': !(0, runtime_js_1.exists)(json, 'code') ? undefined : json['code'],
        'message': !(0, runtime_js_1.exists)(json, 'message') ? undefined : json['message'],
        'permissions': !(0, runtime_js_1.exists)(json, 'permissions') ? undefined : (json['permissions'].map(OrganizationUserPermission_js_1.OrganizationUserPermissionFromJSON)),
    };
}
exports.GetOrganizationsUserPermissionsResponseFromJSONTyped = GetOrganizationsUserPermissionsResponseFromJSONTyped;
function GetOrganizationsUserPermissionsResponseToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'code': value.code,
        'message': value.message,
        'permissions': value.permissions === undefined ? undefined : (value.permissions.map(OrganizationUserPermission_js_1.OrganizationUserPermissionToJSON)),
    };
}
exports.GetOrganizationsUserPermissionsResponseToJSON = GetOrganizationsUserPermissionsResponseToJSON;
