// @kinde-oss/kinde-typescript-sdk/dist-cjs/models/GetOrganizationsUserRolesResponse.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetOrganizationsUserRolesResponseToJSON = exports.GetOrganizationsUserRolesResponseFromJSONTyped = exports.GetOrganizationsUserRolesResponseFromJSON = exports.instanceOfGetOrganizationsUserRolesResponse = void 0;
var runtime_js_1 = require("../runtime.js");
var OrganizationUserRole_js_1 = require("./OrganizationUserRole.js");
/**
 * Check if a given object implements the GetOrganizationsUserRolesResponse interface.
 */
function instanceOfGetOrganizationsUserRolesResponse(value) {
    var isInstance = true;
    return isInstance;
}
exports.instanceOfGetOrganizationsUserRolesResponse = instanceOfGetOrganizationsUserRolesResponse;
function GetOrganizationsUserRolesResponseFromJSON(json) {
    return GetOrganizationsUserRolesResponseFromJSONTyped(json, false);
}
exports.GetOrganizationsUserRolesResponseFromJSON = GetOrganizationsUserRolesResponseFromJSON;
function GetOrganizationsUserRolesResponseFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'code': !(0, runtime_js_1.exists)(json, 'code') ? undefined : json['code'],
        'message': !(0, runtime_js_1.exists)(json, 'message') ? undefined : json['message'],
        'roles': !(0, runtime_js_1.exists)(json, 'roles') ? undefined : (json['roles'].map(OrganizationUserRole_js_1.OrganizationUserRoleFromJSON)),
        'nextToken': !(0, runtime_js_1.exists)(json, 'next_token') ? undefined : json['next_token'],
    };
}
exports.GetOrganizationsUserRolesResponseFromJSONTyped = GetOrganizationsUserRolesResponseFromJSONTyped;
function GetOrganizationsUserRolesResponseToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'code': value.code,
        'message': value.message,
        'roles': value.roles === undefined ? undefined : (value.roles.map(OrganizationUserRole_js_1.OrganizationUserRoleToJSON)),
        'next_token': value.nextToken,
    };
}
exports.GetOrganizationsUserRolesResponseToJSON = GetOrganizationsUserRolesResponseToJSON;
