// @kinde-oss/kinde-typescript-sdk/dist-cjs/models/GetOrganizationsResponse.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetOrganizationsResponseToJSON = exports.GetOrganizationsResponseFromJSONTyped = exports.GetOrganizationsResponseFromJSON = exports.instanceOfGetOrganizationsResponse = void 0;
var runtime_js_1 = require("../runtime.js");
var Organization_js_1 = require("./Organization.js");
/**
 * Check if a given object implements the GetOrganizationsResponse interface.
 */
function instanceOfGetOrganizationsResponse(value) {
    var isInstance = true;
    return isInstance;
}
exports.instanceOfGetOrganizationsResponse = instanceOfGetOrganizationsResponse;
function GetOrganizationsResponseFromJSON(json) {
    return GetOrganizationsResponseFromJSONTyped(json, false);
}
exports.GetOrganizationsResponseFromJSON = GetOrganizationsResponseFromJSON;
function GetOrganizationsResponseFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'code': !(0, runtime_js_1.exists)(json, 'code') ? undefined : json['code'],
        'message': !(0, runtime_js_1.exists)(json, 'message') ? undefined : json['message'],
        'organizations': !(0, runtime_js_1.exists)(json, 'organizations') ? undefined : (json['organizations'].map(Organization_js_1.OrganizationFromJSON)),
        'nextToken': !(0, runtime_js_1.exists)(json, 'next_token') ? undefined : json['next_token'],
    };
}
exports.GetOrganizationsResponseFromJSONTyped = GetOrganizationsResponseFromJSONTyped;
function GetOrganizationsResponseToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'code': value.code,
        'message': value.message,
        'organizations': value.organizations === undefined ? undefined : (value.organizations.map(Organization_js_1.OrganizationToJSON)),
        'next_token': value.nextToken,
    };
}
exports.GetOrganizationsResponseToJSON = GetOrganizationsResponseToJSON;
