// @kinde-oss/kinde-typescript-sdk/dist-cjs/models/GetRolesResponse.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetRolesResponseToJSON = exports.GetRolesResponseFromJSONTyped = exports.GetRolesResponseFromJSON = exports.instanceOfGetRolesResponse = void 0;
var runtime_js_1 = require("../runtime.js");
var Roles_js_1 = require("./Roles.js");
/**
 * Check if a given object implements the GetRolesResponse interface.
 */
function instanceOfGetRolesResponse(value) {
    var isInstance = true;
    return isInstance;
}
exports.instanceOfGetRolesResponse = instanceOfGetRolesResponse;
function GetRolesResponseFromJSON(json) {
    return GetRolesResponseFromJSONTyped(json, false);
}
exports.GetRolesResponseFromJSON = GetRolesResponseFromJSON;
function GetRolesResponseFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'code': !(0, runtime_js_1.exists)(json, 'code') ? undefined : json['code'],
        'message': !(0, runtime_js_1.exists)(json, 'message') ? undefined : json['message'],
        'roles': !(0, runtime_js_1.exists)(json, 'roles') ? undefined : (json['roles'].map(Roles_js_1.RolesFromJSON)),
        'nextToken': !(0, runtime_js_1.exists)(json, 'next_token') ? undefined : json['next_token'],
    };
}
exports.GetRolesResponseFromJSONTyped = GetRolesResponseFromJSONTyped;
function GetRolesResponseToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'code': value.code,
        'message': value.message,
        'roles': value.roles === undefined ? undefined : (value.roles.map(Roles_js_1.RolesToJSON)),
        'next_token': value.nextToken,
    };
}
exports.GetRolesResponseToJSON = GetRolesResponseToJSON;
