// @kinde-oss/kinde-typescript-sdk/dist-cjs/models/GetRedirectCallbackUrlsResponse.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetRedirectCallbackUrlsResponseToJSON = exports.GetRedirectCallbackUrlsResponseFromJSONTyped = exports.GetRedirectCallbackUrlsResponseFromJSON = exports.instanceOfGetRedirectCallbackUrlsResponse = void 0;
var runtime_js_1 = require("../runtime.js");
var RedirectCallbackUrls_js_1 = require("./RedirectCallbackUrls.js");
/**
 * Check if a given object implements the GetRedirectCallbackUrlsResponse interface.
 */
function instanceOfGetRedirectCallbackUrlsResponse(value) {
    var isInstance = true;
    return isInstance;
}
exports.instanceOfGetRedirectCallbackUrlsResponse = instanceOfGetRedirectCallbackUrlsResponse;
function GetRedirectCallbackUrlsResponseFromJSON(json) {
    return GetRedirectCallbackUrlsResponseFromJSONTyped(json, false);
}
exports.GetRedirectCallbackUrlsResponseFromJSON = GetRedirectCallbackUrlsResponseFromJSON;
function GetRedirectCallbackUrlsResponseFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'redirectUrls': !(0, runtime_js_1.exists)(json, 'redirect_urls') ? undefined : (json['redirect_urls'].map(RedirectCallbackUrls_js_1.RedirectCallbackUrlsFromJSON)),
    };
}
exports.GetRedirectCallbackUrlsResponseFromJSONTyped = GetRedirectCallbackUrlsResponseFromJSONTyped;
function GetRedirectCallbackUrlsResponseToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'redirect_urls': value.redirectUrls === undefined ? undefined : (value.redirectUrls.map(RedirectCallbackUrls_js_1.RedirectCallbackUrlsToJSON)),
    };
}
exports.GetRedirectCallbackUrlsResponseToJSON = GetRedirectCallbackUrlsResponseToJSON;
