// @kinde-oss/kinde-typescript-sdk/dist-cjs/models/GetSubscribersResponse.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetSubscribersResponseToJSON = exports.GetSubscribersResponseFromJSONTyped = exports.GetSubscribersResponseFromJSON = exports.instanceOfGetSubscribersResponse = void 0;
var runtime_js_1 = require("../runtime.js");
var SubscribersSubscriber_js_1 = require("./SubscribersSubscriber.js");
/**
 * Check if a given object implements the GetSubscribersResponse interface.
 */
function instanceOfGetSubscribersResponse(value) {
    var isInstance = true;
    return isInstance;
}
exports.instanceOfGetSubscribersResponse = instanceOfGetSubscribersResponse;
function GetSubscribersResponseFromJSON(json) {
    return GetSubscribersResponseFromJSONTyped(json, false);
}
exports.GetSubscribersResponseFromJSON = GetSubscribersResponseFromJSON;
function GetSubscribersResponseFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'code': !(0, runtime_js_1.exists)(json, 'code') ? undefined : json['code'],
        'message': !(0, runtime_js_1.exists)(json, 'message') ? undefined : json['message'],
        'subscribers': !(0, runtime_js_1.exists)(json, 'subscribers') ? undefined : (json['subscribers'].map(SubscribersSubscriber_js_1.SubscribersSubscriberFromJSON)),
        'nextToken': !(0, runtime_js_1.exists)(json, 'next_token') ? undefined : json['next_token'],
    };
}
exports.GetSubscribersResponseFromJSONTyped = GetSubscribersResponseFromJSONTyped;
function GetSubscribersResponseToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'code': value.code,
        'message': value.message,
        'subscribers': value.subscribers === undefined ? undefined : (value.subscribers.map(SubscribersSubscriber_js_1.SubscribersSubscriberToJSON)),
        'next_token': value.nextToken,
    };
}
exports.GetSubscribersResponseToJSON = GetSubscribersResponseToJSON;
