// @kinde-oss/kinde-typescript-sdk/dist-cjs/models/GetSubscriberResponse.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetSubscriberResponseToJSON = exports.GetSubscriberResponseFromJSONTyped = exports.GetSubscriberResponseFromJSON = exports.instanceOfGetSubscriberResponse = void 0;
var runtime_js_1 = require("../runtime.js");
var Subscriber_js_1 = require("./Subscriber.js");
/**
 * Check if a given object implements the GetSubscriberResponse interface.
 */
function instanceOfGetSubscriberResponse(value) {
    var isInstance = true;
    return isInstance;
}
exports.instanceOfGetSubscriberResponse = instanceOfGetSubscriberResponse;
function GetSubscriberResponseFromJSON(json) {
    return GetSubscriberResponseFromJSONTyped(json, false);
}
exports.GetSubscriberResponseFromJSON = GetSubscriberResponseFromJSON;
function GetSubscriberResponseFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'code': !(0, runtime_js_1.exists)(json, 'code') ? undefined : json['code'],
        'message': !(0, runtime_js_1.exists)(json, 'message') ? undefined : json['message'],
        'subscribers': !(0, runtime_js_1.exists)(json, 'subscribers') ? undefined : (json['subscribers'].map(Subscriber_js_1.SubscriberFromJSON)),
    };
}
exports.GetSubscriberResponseFromJSONTyped = GetSubscriberResponseFromJSONTyped;
function GetSubscriberResponseToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'code': value.code,
        'message': value.message,
        'subscribers': value.subscribers === undefined ? undefined : (value.subscribers.map(Subscriber_js_1.SubscriberToJSON)),
    };
}
exports.GetSubscriberResponseToJSON = GetSubscriberResponseToJSON;
