// @kinde-oss/kinde-typescript-sdk/dist-cjs/models/LogoutRedirectUrls.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.LogoutRedirectUrlsToJSON = exports.LogoutRedirectUrlsFromJSONTyped = exports.LogoutRedirectUrlsFromJSON = exports.instanceOfLogoutRedirectUrls = void 0;
var runtime_js_1 = require("../runtime.js");
/**
 * Check if a given object implements the LogoutRedirectUrls interface.
 */
function instanceOfLogoutRedirectUrls(value) {
    var isInstance = true;
    return isInstance;
}
exports.instanceOfLogoutRedirectUrls = instanceOfLogoutRedirectUrls;
function LogoutRedirectUrlsFromJSON(json) {
    return LogoutRedirectUrlsFromJSONTyped(json, false);
}
exports.LogoutRedirectUrlsFromJSON = LogoutRedirectUrlsFromJSON;
function LogoutRedirectUrlsFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'redirectUrls': !(0, runtime_js_1.exists)(json, 'redirect_urls') ? undefined : json['redirect_urls'],
    };
}
exports.LogoutRedirectUrlsFromJSONTyped = LogoutRedirectUrlsFromJSONTyped;
function LogoutRedirectUrlsToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'redirect_urls': value.redirectUrls,
    };
}
exports.LogoutRedirectUrlsToJSON = LogoutRedirectUrlsToJSON;
