// @kinde-oss/kinde-typescript-sdk/dist-cjs/models/OrganizationUserPermission.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.OrganizationUserPermissionToJSON = exports.OrganizationUserPermissionFromJSONTyped = exports.OrganizationUserPermissionFromJSON = exports.instanceOfOrganizationUserPermission = void 0;
var runtime_js_1 = require("../runtime.js");
var OrganizationUserPermissionRolesInner_js_1 = require("./OrganizationUserPermissionRolesInner.js");
/**
 * Check if a given object implements the OrganizationUserPermission interface.
 */
function instanceOfOrganizationUserPermission(value) {
    var isInstance = true;
    return isInstance;
}
exports.instanceOfOrganizationUserPermission = instanceOfOrganizationUserPermission;
function OrganizationUserPermissionFromJSON(json) {
    return OrganizationUserPermissionFromJSONTyped(json, false);
}
exports.OrganizationUserPermissionFromJSON = OrganizationUserPermissionFromJSON;
function OrganizationUserPermissionFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'id': !(0, runtime_js_1.exists)(json, 'id') ? undefined : json['id'],
        'key': !(0, runtime_js_1.exists)(json, 'key') ? undefined : json['key'],
        'name': !(0, runtime_js_1.exists)(json, 'name') ? undefined : json['name'],
        'description': !(0, runtime_js_1.exists)(json, 'description') ? undefined : json['description'],
        'roles': !(0, runtime_js_1.exists)(json, 'roles') ? undefined : (json['roles'].map(OrganizationUserPermissionRolesInner_js_1.OrganizationUserPermissionRolesInnerFromJSON)),
    };
}
exports.OrganizationUserPermissionFromJSONTyped = OrganizationUserPermissionFromJSONTyped;
function OrganizationUserPermissionToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'id': value.id,
        'key': value.key,
        'name': value.name,
        'description': value.description,
        'roles': value.roles === undefined ? undefined : (value.roles.map(OrganizationUserPermissionRolesInner_js_1.OrganizationUserPermissionRolesInnerToJSON)),
    };
}
exports.OrganizationUserPermissionToJSON = OrganizationUserPermissionToJSON;
