// @kinde-oss/kinde-typescript-sdk/dist-cjs/models/OrganizationUserPermissionRolesInner.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.OrganizationUserPermissionRolesInnerToJSON = exports.OrganizationUserPermissionRolesInnerFromJSONTyped = exports.OrganizationUserPermissionRolesInnerFromJSON = exports.instanceOfOrganizationUserPermissionRolesInner = void 0;
var runtime_js_1 = require("../runtime.js");
/**
 * Check if a given object implements the OrganizationUserPermissionRolesInner interface.
 */
function instanceOfOrganizationUserPermissionRolesInner(value) {
    var isInstance = true;
    return isInstance;
}
exports.instanceOfOrganizationUserPermissionRolesInner = instanceOfOrganizationUserPermissionRolesInner;
function OrganizationUserPermissionRolesInnerFromJSON(json) {
    return OrganizationUserPermissionRolesInnerFromJSONTyped(json, false);
}
exports.OrganizationUserPermissionRolesInnerFromJSON = OrganizationUserPermissionRolesInnerFromJSON;
function OrganizationUserPermissionRolesInnerFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'id': !(0, runtime_js_1.exists)(json, 'id') ? undefined : json['id'],
        'key': !(0, runtime_js_1.exists)(json, 'key') ? undefined : json['key'],
    };
}
exports.OrganizationUserPermissionRolesInnerFromJSONTyped = OrganizationUserPermissionRolesInnerFromJSONTyped;
function OrganizationUserPermissionRolesInnerToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'id': value.id,
        'key': value.key,
    };
}
exports.OrganizationUserPermissionRolesInnerToJSON = OrganizationUserPermissionRolesInnerToJSON;
