// @kinde-oss/kinde-typescript-sdk/dist-cjs/models/OrganizationUserRole.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.OrganizationUserRoleToJSON = exports.OrganizationUserRoleFromJSONTyped = exports.OrganizationUserRoleFromJSON = exports.instanceOfOrganizationUserRole = void 0;
var runtime_js_1 = require("../runtime.js");
/**
 * Check if a given object implements the OrganizationUserRole interface.
 */
function instanceOfOrganizationUserRole(value) {
    var isInstance = true;
    return isInstance;
}
exports.instanceOfOrganizationUserRole = instanceOfOrganizationUserRole;
function OrganizationUserRoleFromJSON(json) {
    return OrganizationUserRoleFromJSONTyped(json, false);
}
exports.OrganizationUserRoleFromJSON = OrganizationUserRoleFromJSON;
function OrganizationUserRoleFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'id': !(0, runtime_js_1.exists)(json, 'id') ? undefined : json['id'],
        'key': !(0, runtime_js_1.exists)(json, 'key') ? undefined : json['key'],
        'name': !(0, runtime_js_1.exists)(json, 'name') ? undefined : json['name'],
    };
}
exports.OrganizationUserRoleFromJSONTyped = OrganizationUserRoleFromJSONTyped;
function OrganizationUserRoleToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'id': value.id,
        'key': value.key,
        'name': value.name,
    };
}
exports.OrganizationUserRoleToJSON = OrganizationUserRoleToJSON;
