// @kinde-oss/kinde-typescript-sdk/dist-cjs/models/OrganizationUserRolePermissionsPermissions.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.OrganizationUserRolePermissionsPermissionsToJSON = exports.OrganizationUserRolePermissionsPermissionsFromJSONTyped = exports.OrganizationUserRolePermissionsPermissionsFromJSON = exports.instanceOfOrganizationUserRolePermissionsPermissions = void 0;
var runtime_js_1 = require("../runtime.js");
/**
 * Check if a given object implements the OrganizationUserRolePermissionsPermissions interface.
 */
function instanceOfOrganizationUserRolePermissionsPermissions(value) {
    var isInstance = true;
    return isInstance;
}
exports.instanceOfOrganizationUserRolePermissionsPermissions = instanceOfOrganizationUserRolePermissionsPermissions;
function OrganizationUserRolePermissionsPermissionsFromJSON(json) {
    return OrganizationUserRolePermissionsPermissionsFromJSONTyped(json, false);
}
exports.OrganizationUserRolePermissionsPermissionsFromJSON = OrganizationUserRolePermissionsPermissionsFromJSON;
function OrganizationUserRolePermissionsPermissionsFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'key': !(0, runtime_js_1.exists)(json, 'key') ? undefined : json['key'],
    };
}
exports.OrganizationUserRolePermissionsPermissionsFromJSONTyped = OrganizationUserRolePermissionsPermissionsFromJSONTyped;
function OrganizationUserRolePermissionsPermissionsToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'key': value.key,
    };
}
exports.OrganizationUserRolePermissionsPermissionsToJSON = OrganizationUserRolePermissionsPermissionsToJSON;
