// @kinde-oss/kinde-typescript-sdk/dist-cjs/models/OrganizationUserRolePermissions.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.OrganizationUserRolePermissionsToJSON = exports.OrganizationUserRolePermissionsFromJSONTyped = exports.OrganizationUserRolePermissionsFromJSON = exports.instanceOfOrganizationUserRolePermissions = void 0;
var runtime_js_1 = require("../runtime.js");
var OrganizationUserRolePermissionsPermissions_js_1 = require("./OrganizationUserRolePermissionsPermissions.js");
/**
 * Check if a given object implements the OrganizationUserRolePermissions interface.
 */
function instanceOfOrganizationUserRolePermissions(value) {
    var isInstance = true;
    return isInstance;
}
exports.instanceOfOrganizationUserRolePermissions = instanceOfOrganizationUserRolePermissions;
function OrganizationUserRolePermissionsFromJSON(json) {
    return OrganizationUserRolePermissionsFromJSONTyped(json, false);
}
exports.OrganizationUserRolePermissionsFromJSON = OrganizationUserRolePermissionsFromJSON;
function OrganizationUserRolePermissionsFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'id': !(0, runtime_js_1.exists)(json, 'id') ? undefined : json['id'],
        'role': !(0, runtime_js_1.exists)(json, 'role') ? undefined : json['role'],
        'permissions': !(0, runtime_js_1.exists)(json, 'permissions') ? undefined : (0, OrganizationUserRolePermissionsPermissions_js_1.OrganizationUserRolePermissionsPermissionsFromJSON)(json['permissions']),
    };
}
exports.OrganizationUserRolePermissionsFromJSONTyped = OrganizationUserRolePermissionsFromJSONTyped;
function OrganizationUserRolePermissionsToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'id': value.id,
        'role': value.role,
        'permissions': (0, OrganizationUserRolePermissionsPermissions_js_1.OrganizationUserRolePermissionsPermissionsToJSON)(value.permissions),
    };
}
exports.OrganizationUserRolePermissionsToJSON = OrganizationUserRolePermissionsToJSON;
