// @kinde-oss/kinde-typescript-sdk/dist-cjs/models/Organization.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.OrganizationToJSON = exports.OrganizationFromJSONTyped = exports.OrganizationFromJSON = exports.instanceOfOrganization = void 0;
var runtime_js_1 = require("../runtime.js");
/**
 * Check if a given object implements the Organization interface.
 */
function instanceOfOrganization(value) {
    var isInstance = true;
    return isInstance;
}
exports.instanceOfOrganization = instanceOfOrganization;
function OrganizationFromJSON(json) {
    return OrganizationFromJSONTyped(json, false);
}
exports.OrganizationFromJSON = OrganizationFromJSON;
function OrganizationFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'code': !(0, runtime_js_1.exists)(json, 'code') ? undefined : json['code'],
        'name': !(0, runtime_js_1.exists)(json, 'name') ? undefined : json['name'],
        'isDefault': !(0, runtime_js_1.exists)(json, 'is_default') ? undefined : json['is_default'],
        'externalId': !(0, runtime_js_1.exists)(json, 'external_id') ? undefined : json['external_id'],
    };
}
exports.OrganizationFromJSONTyped = OrganizationFromJSONTyped;
function OrganizationToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'code': value.code,
        'name': value.name,
        'is_default': value.isDefault,
        'external_id': value.externalId,
    };
}
exports.OrganizationToJSON = OrganizationToJSON;
