// @kinde-oss/kinde-typescript-sdk/dist-cjs/models/RedirectCallbackUrls.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.RedirectCallbackUrlsToJSON = exports.RedirectCallbackUrlsFromJSONTyped = exports.RedirectCallbackUrlsFromJSON = exports.instanceOfRedirectCallbackUrls = void 0;
var runtime_js_1 = require("../runtime.js");
/**
 * Check if a given object implements the RedirectCallbackUrls interface.
 */
function instanceOfRedirectCallbackUrls(value) {
    var isInstance = true;
    return isInstance;
}
exports.instanceOfRedirectCallbackUrls = instanceOfRedirectCallbackUrls;
function RedirectCallbackUrlsFromJSON(json) {
    return RedirectCallbackUrlsFromJSONTyped(json, false);
}
exports.RedirectCallbackUrlsFromJSON = RedirectCallbackUrlsFromJSON;
function RedirectCallbackUrlsFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'redirectUrls': !(0, runtime_js_1.exists)(json, 'redirect_urls') ? undefined : json['redirect_urls'],
    };
}
exports.RedirectCallbackUrlsFromJSONTyped = RedirectCallbackUrlsFromJSONTyped;
function RedirectCallbackUrlsToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'redirect_urls': value.redirectUrls,
    };
}
exports.RedirectCallbackUrlsToJSON = RedirectCallbackUrlsToJSON;
