// @kinde-oss/kinde-typescript-sdk/dist-cjs/models/ReplaceRedirectCallbackURLsRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ReplaceRedirectCallbackURLsRequestToJSON = exports.ReplaceRedirectCallbackURLsRequestFromJSONTyped = exports.ReplaceRedirectCallbackURLsRequestFromJSON = exports.instanceOfReplaceRedirectCallbackURLsRequest = void 0;
var runtime_js_1 = require("../runtime.js");
/**
 * Check if a given object implements the ReplaceRedirectCallbackURLsRequest interface.
 */
function instanceOfReplaceRedirectCallbackURLsRequest(value) {
    var isInstance = true;
    return isInstance;
}
exports.instanceOfReplaceRedirectCallbackURLsRequest = instanceOfReplaceRedirectCallbackURLsRequest;
function ReplaceRedirectCallbackURLsRequestFromJSON(json) {
    return ReplaceRedirectCallbackURLsRequestFromJSONTyped(json, false);
}
exports.ReplaceRedirectCallbackURLsRequestFromJSON = ReplaceRedirectCallbackURLsRequestFromJSON;
function ReplaceRedirectCallbackURLsRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'urls': !(0, runtime_js_1.exists)(json, 'urls') ? undefined : json['urls'],
    };
}
exports.ReplaceRedirectCallbackURLsRequestFromJSONTyped = ReplaceRedirectCallbackURLsRequestFromJSONTyped;
function ReplaceRedirectCallbackURLsRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'urls': value.urls,
    };
}
exports.ReplaceRedirectCallbackURLsRequestToJSON = ReplaceRedirectCallbackURLsRequestToJSON;
