// @kinde-oss/kinde-typescript-sdk/dist-cjs/models/ReplaceLogoutRedirectURLsRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ReplaceLogoutRedirectURLsRequestToJSON = exports.ReplaceLogoutRedirectURLsRequestFromJSONTyped = exports.ReplaceLogoutRedirectURLsRequestFromJSON = exports.instanceOfReplaceLogoutRedirectURLsRequest = void 0;
var runtime_js_1 = require("../runtime.js");
/**
 * Check if a given object implements the ReplaceLogoutRedirectURLsRequest interface.
 */
function instanceOfReplaceLogoutRedirectURLsRequest(value) {
    var isInstance = true;
    return isInstance;
}
exports.instanceOfReplaceLogoutRedirectURLsRequest = instanceOfReplaceLogoutRedirectURLsRequest;
function ReplaceLogoutRedirectURLsRequestFromJSON(json) {
    return ReplaceLogoutRedirectURLsRequestFromJSONTyped(json, false);
}
exports.ReplaceLogoutRedirectURLsRequestFromJSON = ReplaceLogoutRedirectURLsRequestFromJSON;
function ReplaceLogoutRedirectURLsRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'urls': !(0, runtime_js_1.exists)(json, 'urls') ? undefined : json['urls'],
    };
}
exports.ReplaceLogoutRedirectURLsRequestFromJSONTyped = ReplaceLogoutRedirectURLsRequestFromJSONTyped;
function ReplaceLogoutRedirectURLsRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'urls': value.urls,
    };
}
exports.ReplaceLogoutRedirectURLsRequestToJSON = ReplaceLogoutRedirectURLsRequestToJSON;
