// @kinde-oss/kinde-typescript-sdk/dist-cjs/models/Roles.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.RolesToJSON = exports.RolesFromJSONTyped = exports.RolesFromJSON = exports.instanceOfRoles = void 0;
var runtime_js_1 = require("../runtime.js");
/**
 * Check if a given object implements the Roles interface.
 */
function instanceOfRoles(value) {
    var isInstance = true;
    return isInstance;
}
exports.instanceOfRoles = instanceOfRoles;
function RolesFromJSON(json) {
    return RolesFromJSONTyped(json, false);
}
exports.RolesFromJSON = RolesFromJSON;
function RolesFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'id': !(0, runtime_js_1.exists)(json, 'id') ? undefined : json['id'],
        'key': !(0, runtime_js_1.exists)(json, 'key') ? undefined : json['key'],
        'name': !(0, runtime_js_1.exists)(json, 'name') ? undefined : json['name'],
        'description': !(0, runtime_js_1.exists)(json, 'description') ? undefined : json['description'],
    };
}
exports.RolesFromJSONTyped = RolesFromJSONTyped;
function RolesToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'id': value.id,
        'key': value.key,
        'name': value.name,
        'description': value.description,
    };
}
exports.RolesToJSON = RolesToJSON;
