// @kinde-oss/kinde-typescript-sdk/dist-cjs/models/SubscribersSubscriber.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SubscribersSubscriberToJSON = exports.SubscribersSubscriberFromJSONTyped = exports.SubscribersSubscriberFromJSON = exports.instanceOfSubscribersSubscriber = void 0;
var runtime_js_1 = require("../runtime.js");
/**
 * Check if a given object implements the SubscribersSubscriber interface.
 */
function instanceOfSubscribersSubscriber(value) {
    var isInstance = true;
    return isInstance;
}
exports.instanceOfSubscribersSubscriber = instanceOfSubscribersSubscriber;
function SubscribersSubscriberFromJSON(json) {
    return SubscribersSubscriberFromJSONTyped(json, false);
}
exports.SubscribersSubscriberFromJSON = SubscribersSubscriberFromJSON;
function SubscribersSubscriberFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'id': !(0, runtime_js_1.exists)(json, 'id') ? undefined : json['id'],
        'email': !(0, runtime_js_1.exists)(json, 'email') ? undefined : json['email'],
        'fullName': !(0, runtime_js_1.exists)(json, 'full_name') ? undefined : json['full_name'],
        'firstName': !(0, runtime_js_1.exists)(json, 'first_name') ? undefined : json['first_name'],
        'lastName': !(0, runtime_js_1.exists)(json, 'last_name') ? undefined : json['last_name'],
    };
}
exports.SubscribersSubscriberFromJSONTyped = SubscribersSubscriberFromJSONTyped;
function SubscribersSubscriberToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'id': value.id,
        'email': value.email,
        'full_name': value.fullName,
        'first_name': value.firstName,
        'last_name': value.lastName,
    };
}
exports.SubscribersSubscriberToJSON = SubscribersSubscriberToJSON;
