// @kinde-oss/kinde-typescript-sdk/dist-cjs/models/Subscriber.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SubscriberToJSON = exports.SubscriberFromJSONTyped = exports.SubscriberFromJSON = exports.instanceOfSubscriber = void 0;
var runtime_js_1 = require("../runtime.js");
/**
 * Check if a given object implements the Subscriber interface.
 */
function instanceOfSubscriber(value) {
    var isInstance = true;
    return isInstance;
}
exports.instanceOfSubscriber = instanceOfSubscriber;
function SubscriberFromJSON(json) {
    return SubscriberFromJSONTyped(json, false);
}
exports.SubscriberFromJSON = SubscriberFromJSON;
function SubscriberFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'id': !(0, runtime_js_1.exists)(json, 'id') ? undefined : json['id'],
        'preferredEmail': !(0, runtime_js_1.exists)(json, 'preferred_email') ? undefined : json['preferred_email'],
        'firstName': !(0, runtime_js_1.exists)(json, 'first_name') ? undefined : json['first_name'],
        'lastName': !(0, runtime_js_1.exists)(json, 'last_name') ? undefined : json['last_name'],
    };
}
exports.SubscriberFromJSONTyped = SubscriberFromJSONTyped;
function SubscriberToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'id': value.id,
        'preferred_email': value.preferredEmail,
        'first_name': value.firstName,
        'last_name': value.lastName,
    };
}
exports.SubscriberToJSON = SubscriberToJSON;
