// @kinde-oss/kinde-typescript-sdk/dist-cjs/models/SuccessResponse.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SuccessResponseToJSON = exports.SuccessResponseFromJSONTyped = exports.SuccessResponseFromJSON = exports.instanceOfSuccessResponse = void 0;
var runtime_js_1 = require("../runtime.js");
/**
 * Check if a given object implements the SuccessResponse interface.
 */
function instanceOfSuccessResponse(value) {
    var isInstance = true;
    return isInstance;
}
exports.instanceOfSuccessResponse = instanceOfSuccessResponse;
function SuccessResponseFromJSON(json) {
    return SuccessResponseFromJSONTyped(json, false);
}
exports.SuccessResponseFromJSON = SuccessResponseFromJSON;
function SuccessResponseFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'message': !(0, runtime_js_1.exists)(json, 'message') ? undefined : json['message'],
        'code': !(0, runtime_js_1.exists)(json, 'code') ? undefined : json['code'],
    };
}
exports.SuccessResponseFromJSONTyped = SuccessResponseFromJSONTyped;
function SuccessResponseToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'message': value.message,
        'code': value.code,
    };
}
exports.SuccessResponseToJSON = SuccessResponseToJSON;
