// @kinde-oss/kinde-typescript-sdk/dist-cjs/models/TokenIntrospect.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TokenIntrospectToJSON = exports.TokenIntrospectFromJSONTyped = exports.TokenIntrospectFromJSON = exports.instanceOfTokenIntrospect = void 0;
var runtime_js_1 = require("../runtime.js");
/**
 * Check if a given object implements the TokenIntrospect interface.
 */
function instanceOfTokenIntrospect(value) {
    var isInstance = true;
    return isInstance;
}
exports.instanceOfTokenIntrospect = instanceOfTokenIntrospect;
function TokenIntrospectFromJSON(json) {
    return TokenIntrospectFromJSONTyped(json, false);
}
exports.TokenIntrospectFromJSON = TokenIntrospectFromJSON;
function TokenIntrospectFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'active': !(0, runtime_js_1.exists)(json, 'active') ? undefined : json['active'],
        'aud': !(0, runtime_js_1.exists)(json, 'aud') ? undefined : json['aud'],
        'clientId': !(0, runtime_js_1.exists)(json, 'client_id') ? undefined : json['client_id'],
        'exp': !(0, runtime_js_1.exists)(json, 'exp') ? undefined : json['exp'],
        'iat': !(0, runtime_js_1.exists)(json, 'iat') ? undefined : json['iat'],
    };
}
exports.TokenIntrospectFromJSONTyped = TokenIntrospectFromJSONTyped;
function TokenIntrospectToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'active': value.active,
        'aud': value.aud,
        'client_id': value.clientId,
        'exp': value.exp,
        'iat': value.iat,
    };
}
exports.TokenIntrospectToJSON = TokenIntrospectToJSON;
