// @kinde-oss/kinde-typescript-sdk/dist-cjs/models/TokenErrorResponse.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TokenErrorResponseToJSON = exports.TokenErrorResponseFromJSONTyped = exports.TokenErrorResponseFromJSON = exports.instanceOfTokenErrorResponse = void 0;
var runtime_js_1 = require("../runtime.js");
/**
 * Check if a given object implements the TokenErrorResponse interface.
 */
function instanceOfTokenErrorResponse(value) {
    var isInstance = true;
    return isInstance;
}
exports.instanceOfTokenErrorResponse = instanceOfTokenErrorResponse;
function TokenErrorResponseFromJSON(json) {
    return TokenErrorResponseFromJSONTyped(json, false);
}
exports.TokenErrorResponseFromJSON = TokenErrorResponseFromJSON;
function TokenErrorResponseFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'error': !(0, runtime_js_1.exists)(json, 'error') ? undefined : json['error'],
        'errorDescription': !(0, runtime_js_1.exists)(json, 'error_description') ? undefined : json['error_description'],
    };
}
exports.TokenErrorResponseFromJSONTyped = TokenErrorResponseFromJSONTyped;
function TokenErrorResponseToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'error': value.error,
        'error_description': value.errorDescription,
    };
}
exports.TokenErrorResponseToJSON = TokenErrorResponseToJSON;
