// @kinde-oss/kinde-typescript-sdk/dist-cjs/models/UpdateApplicationRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdateApplicationRequestToJSON = exports.UpdateApplicationRequestFromJSONTyped = exports.UpdateApplicationRequestFromJSON = exports.instanceOfUpdateApplicationRequest = void 0;
var runtime_js_1 = require("../runtime.js");
/**
 * Check if a given object implements the UpdateApplicationRequest interface.
 */
function instanceOfUpdateApplicationRequest(value) {
    var isInstance = true;
    return isInstance;
}
exports.instanceOfUpdateApplicationRequest = instanceOfUpdateApplicationRequest;
function UpdateApplicationRequestFromJSON(json) {
    return UpdateApplicationRequestFromJSONTyped(json, false);
}
exports.UpdateApplicationRequestFromJSON = UpdateApplicationRequestFromJSON;
function UpdateApplicationRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': !(0, runtime_js_1.exists)(json, 'name') ? undefined : json['name'],
        'languageKey': !(0, runtime_js_1.exists)(json, 'language_key') ? undefined : json['language_key'],
        'logoutUris': !(0, runtime_js_1.exists)(json, 'logout_uris') ? undefined : json['logout_uris'],
        'redirectUris': !(0, runtime_js_1.exists)(json, 'redirect_uris') ? undefined : json['redirect_uris'],
    };
}
exports.UpdateApplicationRequestFromJSONTyped = UpdateApplicationRequestFromJSONTyped;
function UpdateApplicationRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'language_key': value.languageKey,
        'logout_uris': value.logoutUris,
        'redirect_uris': value.redirectUris,
    };
}
exports.UpdateApplicationRequestToJSON = UpdateApplicationRequestToJSON;
