// @kinde-oss/kinde-typescript-sdk/dist-cjs/models/UpdateOrganizationUsersRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdateOrganizationUsersRequestToJSON = exports.UpdateOrganizationUsersRequestFromJSONTyped = exports.UpdateOrganizationUsersRequestFromJSON = exports.instanceOfUpdateOrganizationUsersRequest = void 0;
var runtime_js_1 = require("../runtime.js");
var UpdateOrganizationUsersRequestUsersInner_js_1 = require("./UpdateOrganizationUsersRequestUsersInner.js");
/**
 * Check if a given object implements the UpdateOrganizationUsersRequest interface.
 */
function instanceOfUpdateOrganizationUsersRequest(value) {
    var isInstance = true;
    return isInstance;
}
exports.instanceOfUpdateOrganizationUsersRequest = instanceOfUpdateOrganizationUsersRequest;
function UpdateOrganizationUsersRequestFromJSON(json) {
    return UpdateOrganizationUsersRequestFromJSONTyped(json, false);
}
exports.UpdateOrganizationUsersRequestFromJSON = UpdateOrganizationUsersRequestFromJSON;
function UpdateOrganizationUsersRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'users': !(0, runtime_js_1.exists)(json, 'users') ? undefined : (json['users'].map(UpdateOrganizationUsersRequestUsersInner_js_1.UpdateOrganizationUsersRequestUsersInnerFromJSON)),
    };
}
exports.UpdateOrganizationUsersRequestFromJSONTyped = UpdateOrganizationUsersRequestFromJSONTyped;
function UpdateOrganizationUsersRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'users': value.users === undefined ? undefined : (value.users.map(UpdateOrganizationUsersRequestUsersInner_js_1.UpdateOrganizationUsersRequestUsersInnerToJSON)),
    };
}
exports.UpdateOrganizationUsersRequestToJSON = UpdateOrganizationUsersRequestToJSON;
