// @kinde-oss/kinde-typescript-sdk/dist-cjs/models/UpdateAPIApplicationsRequestApplicationsInner.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdateAPIApplicationsRequestApplicationsInnerToJSON = exports.UpdateAPIApplicationsRequestApplicationsInnerFromJSONTyped = exports.UpdateAPIApplicationsRequestApplicationsInnerFromJSON = exports.instanceOfUpdateAPIApplicationsRequestApplicationsInner = void 0;
var runtime_js_1 = require("../runtime.js");
/**
 * Check if a given object implements the UpdateAPIApplicationsRequestApplicationsInner interface.
 */
function instanceOfUpdateAPIApplicationsRequestApplicationsInner(value) {
    var isInstance = true;
    isInstance = isInstance && "id" in value;
    return isInstance;
}
exports.instanceOfUpdateAPIApplicationsRequestApplicationsInner = instanceOfUpdateAPIApplicationsRequestApplicationsInner;
function UpdateAPIApplicationsRequestApplicationsInnerFromJSON(json) {
    return UpdateAPIApplicationsRequestApplicationsInnerFromJSONTyped(json, false);
}
exports.UpdateAPIApplicationsRequestApplicationsInnerFromJSON = UpdateAPIApplicationsRequestApplicationsInnerFromJSON;
function UpdateAPIApplicationsRequestApplicationsInnerFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'id': json['id'],
        'operation': !(0, runtime_js_1.exists)(json, 'operation') ? undefined : json['operation'],
    };
}
exports.UpdateAPIApplicationsRequestApplicationsInnerFromJSONTyped = UpdateAPIApplicationsRequestApplicationsInnerFromJSONTyped;
function UpdateAPIApplicationsRequestApplicationsInnerToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'id': value.id,
        'operation': value.operation,
    };
}
exports.UpdateAPIApplicationsRequestApplicationsInnerToJSON = UpdateAPIApplicationsRequestApplicationsInnerToJSON;
