// @kinde-oss/kinde-typescript-sdk/dist-cjs/models/UpdateRolePermissionsRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdateRolePermissionsRequestToJSON = exports.UpdateRolePermissionsRequestFromJSONTyped = exports.UpdateRolePermissionsRequestFromJSON = exports.instanceOfUpdateRolePermissionsRequest = void 0;
var runtime_js_1 = require("../runtime.js");
var UpdateRolePermissionsRequestPermissionsInner_js_1 = require("./UpdateRolePermissionsRequestPermissionsInner.js");
/**
 * Check if a given object implements the UpdateRolePermissionsRequest interface.
 */
function instanceOfUpdateRolePermissionsRequest(value) {
    var isInstance = true;
    return isInstance;
}
exports.instanceOfUpdateRolePermissionsRequest = instanceOfUpdateRolePermissionsRequest;
function UpdateRolePermissionsRequestFromJSON(json) {
    return UpdateRolePermissionsRequestFromJSONTyped(json, false);
}
exports.UpdateRolePermissionsRequestFromJSON = UpdateRolePermissionsRequestFromJSON;
function UpdateRolePermissionsRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'permissions': !(0, runtime_js_1.exists)(json, 'permissions') ? undefined : (json['permissions'].map(UpdateRolePermissionsRequestPermissionsInner_js_1.UpdateRolePermissionsRequestPermissionsInnerFromJSON)),
    };
}
exports.UpdateRolePermissionsRequestFromJSONTyped = UpdateRolePermissionsRequestFromJSONTyped;
function UpdateRolePermissionsRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'permissions': value.permissions === undefined ? undefined : (value.permissions.map(UpdateRolePermissionsRequestPermissionsInner_js_1.UpdateRolePermissionsRequestPermissionsInnerToJSON)),
    };
}
exports.UpdateRolePermissionsRequestToJSON = UpdateRolePermissionsRequestToJSON;
