// @kinde-oss/kinde-typescript-sdk/dist-cjs/models/UpdateRolePermissionsResponse.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdateRolePermissionsResponseToJSON = exports.UpdateRolePermissionsResponseFromJSONTyped = exports.UpdateRolePermissionsResponseFromJSON = exports.instanceOfUpdateRolePermissionsResponse = void 0;
var runtime_js_1 = require("../runtime.js");
/**
 * Check if a given object implements the UpdateRolePermissionsResponse interface.
 */
function instanceOfUpdateRolePermissionsResponse(value) {
    var isInstance = true;
    return isInstance;
}
exports.instanceOfUpdateRolePermissionsResponse = instanceOfUpdateRolePermissionsResponse;
function UpdateRolePermissionsResponseFromJSON(json) {
    return UpdateRolePermissionsResponseFromJSONTyped(json, false);
}
exports.UpdateRolePermissionsResponseFromJSON = UpdateRolePermissionsResponseFromJSON;
function UpdateRolePermissionsResponseFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'code': !(0, runtime_js_1.exists)(json, 'code') ? undefined : json['code'],
        'message': !(0, runtime_js_1.exists)(json, 'message') ? undefined : json['message'],
        'permissionsAdded': !(0, runtime_js_1.exists)(json, 'permissions_added') ? undefined : json['permissions_added'],
        'permissionsRemoved': !(0, runtime_js_1.exists)(json, 'permissions_removed') ? undefined : json['permissions_removed'],
    };
}
exports.UpdateRolePermissionsResponseFromJSONTyped = UpdateRolePermissionsResponseFromJSONTyped;
function UpdateRolePermissionsResponseToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'code': value.code,
        'message': value.message,
        'permissions_added': value.permissionsAdded,
        'permissions_removed': value.permissionsRemoved,
    };
}
exports.UpdateRolePermissionsResponseToJSON = UpdateRolePermissionsResponseToJSON;
