// @kinde-oss/kinde-typescript-sdk/dist-cjs/models/UpdateAPIApplicationsRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdateAPIApplicationsRequestToJSON = exports.UpdateAPIApplicationsRequestFromJSONTyped = exports.UpdateAPIApplicationsRequestFromJSON = exports.instanceOfUpdateAPIApplicationsRequest = void 0;
var UpdateAPIApplicationsRequestApplicationsInner_js_1 = require("./UpdateAPIApplicationsRequestApplicationsInner.js");
/**
 * Check if a given object implements the UpdateAPIApplicationsRequest interface.
 */
function instanceOfUpdateAPIApplicationsRequest(value) {
    var isInstance = true;
    isInstance = isInstance && "applications" in value;
    return isInstance;
}
exports.instanceOfUpdateAPIApplicationsRequest = instanceOfUpdateAPIApplicationsRequest;
function UpdateAPIApplicationsRequestFromJSON(json) {
    return UpdateAPIApplicationsRequestFromJSONTyped(json, false);
}
exports.UpdateAPIApplicationsRequestFromJSON = UpdateAPIApplicationsRequestFromJSON;
function UpdateAPIApplicationsRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'applications': (json['applications'].map(UpdateAPIApplicationsRequestApplicationsInner_js_1.UpdateAPIApplicationsRequestApplicationsInnerFromJSON)),
    };
}
exports.UpdateAPIApplicationsRequestFromJSONTyped = UpdateAPIApplicationsRequestFromJSONTyped;
function UpdateAPIApplicationsRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'applications': (value.applications.map(UpdateAPIApplicationsRequestApplicationsInner_js_1.UpdateAPIApplicationsRequestApplicationsInnerToJSON)),
    };
}
exports.UpdateAPIApplicationsRequestToJSON = UpdateAPIApplicationsRequestToJSON;
