// @kinde-oss/kinde-typescript-sdk/dist-cjs/models/UpdateEnvironementFeatureFlagOverrideRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdateEnvironementFeatureFlagOverrideRequestToJSON = exports.UpdateEnvironementFeatureFlagOverrideRequestFromJSONTyped = exports.UpdateEnvironementFeatureFlagOverrideRequestFromJSON = exports.instanceOfUpdateEnvironementFeatureFlagOverrideRequest = void 0;
/**
 * Check if a given object implements the UpdateEnvironementFeatureFlagOverrideRequest interface.
 */
function instanceOfUpdateEnvironementFeatureFlagOverrideRequest(value) {
    var isInstance = true;
    isInstance = isInstance && "value" in value;
    return isInstance;
}
exports.instanceOfUpdateEnvironementFeatureFlagOverrideRequest = instanceOfUpdateEnvironementFeatureFlagOverrideRequest;
function UpdateEnvironementFeatureFlagOverrideRequestFromJSON(json) {
    return UpdateEnvironementFeatureFlagOverrideRequestFromJSONTyped(json, false);
}
exports.UpdateEnvironementFeatureFlagOverrideRequestFromJSON = UpdateEnvironementFeatureFlagOverrideRequestFromJSON;
function UpdateEnvironementFeatureFlagOverrideRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'value': json['value'],
    };
}
exports.UpdateEnvironementFeatureFlagOverrideRequestFromJSONTyped = UpdateEnvironementFeatureFlagOverrideRequestFromJSONTyped;
function UpdateEnvironementFeatureFlagOverrideRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'value': value.value,
    };
}
exports.UpdateEnvironementFeatureFlagOverrideRequestToJSON = UpdateEnvironementFeatureFlagOverrideRequestToJSON;
