// @kinde-oss/kinde-typescript-sdk/dist-cjs/models/UpdateUserResponse.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdateUserResponseToJSON = exports.UpdateUserResponseFromJSONTyped = exports.UpdateUserResponseFromJSON = exports.instanceOfUpdateUserResponse = void 0;
var runtime_js_1 = require("../runtime.js");
/**
 * Check if a given object implements the UpdateUserResponse interface.
 */
function instanceOfUpdateUserResponse(value) {
    var isInstance = true;
    return isInstance;
}
exports.instanceOfUpdateUserResponse = instanceOfUpdateUserResponse;
function UpdateUserResponseFromJSON(json) {
    return UpdateUserResponseFromJSONTyped(json, false);
}
exports.UpdateUserResponseFromJSON = UpdateUserResponseFromJSON;
function UpdateUserResponseFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'id': !(0, runtime_js_1.exists)(json, 'id') ? undefined : json['id'],
        'givenName': !(0, runtime_js_1.exists)(json, 'given_name') ? undefined : json['given_name'],
        'familyName': !(0, runtime_js_1.exists)(json, 'family_name') ? undefined : json['family_name'],
        'email': !(0, runtime_js_1.exists)(json, 'email') ? undefined : json['email'],
        'isSuspended': !(0, runtime_js_1.exists)(json, 'is_suspended') ? undefined : json['is_suspended'],
        'isPasswordResetRequested': !(0, runtime_js_1.exists)(json, 'is_password_reset_requested') ? undefined : json['is_password_reset_requested'],
        'picture': !(0, runtime_js_1.exists)(json, 'picture') ? undefined : json['picture'],
    };
}
exports.UpdateUserResponseFromJSONTyped = UpdateUserResponseFromJSONTyped;
function UpdateUserResponseToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'id': value.id,
        'given_name': value.givenName,
        'family_name': value.familyName,
        'email': value.email,
        'is_suspended': value.isSuspended,
        'is_password_reset_requested': value.isPasswordResetRequested,
        'picture': value.picture,
    };
}
exports.UpdateUserResponseToJSON = UpdateUserResponseToJSON;
