// @kinde-oss/kinde-typescript-sdk/dist-cjs/models/UpdateOrganizationUsersResponse.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdateOrganizationUsersResponseToJSON = exports.UpdateOrganizationUsersResponseFromJSONTyped = exports.UpdateOrganizationUsersResponseFromJSON = exports.instanceOfUpdateOrganizationUsersResponse = void 0;
var runtime_js_1 = require("../runtime.js");
/**
 * Check if a given object implements the UpdateOrganizationUsersResponse interface.
 */
function instanceOfUpdateOrganizationUsersResponse(value) {
    var isInstance = true;
    return isInstance;
}
exports.instanceOfUpdateOrganizationUsersResponse = instanceOfUpdateOrganizationUsersResponse;
function UpdateOrganizationUsersResponseFromJSON(json) {
    return UpdateOrganizationUsersResponseFromJSONTyped(json, false);
}
exports.UpdateOrganizationUsersResponseFromJSON = UpdateOrganizationUsersResponseFromJSON;
function UpdateOrganizationUsersResponseFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'message': !(0, runtime_js_1.exists)(json, 'message') ? undefined : json['message'],
        'usersAdded': !(0, runtime_js_1.exists)(json, 'users_added') ? undefined : json['users_added'],
        'usersUpdated': !(0, runtime_js_1.exists)(json, 'users_updated') ? undefined : json['users_updated'],
        'usersRemoved': !(0, runtime_js_1.exists)(json, 'users_removed') ? undefined : json['users_removed'],
    };
}
exports.UpdateOrganizationUsersResponseFromJSONTyped = UpdateOrganizationUsersResponseFromJSONTyped;
function UpdateOrganizationUsersResponseToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'message': value.message,
        'users_added': value.usersAdded,
        'users_updated': value.usersUpdated,
        'users_removed': value.usersRemoved,
    };
}
exports.UpdateOrganizationUsersResponseToJSON = UpdateOrganizationUsersResponseToJSON;
