// @kinde-oss/kinde-typescript-sdk/dist-cjs/models/UpdateRolesRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdateRolesRequestToJSON = exports.UpdateRolesRequestFromJSONTyped = exports.UpdateRolesRequestFromJSON = exports.instanceOfUpdateRolesRequest = void 0;
var runtime_js_1 = require("../runtime.js");
/**
 * Check if a given object implements the UpdateRolesRequest interface.
 */
function instanceOfUpdateRolesRequest(value) {
    var isInstance = true;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "key" in value;
    return isInstance;
}
exports.instanceOfUpdateRolesRequest = instanceOfUpdateRolesRequest;
function UpdateRolesRequestFromJSON(json) {
    return UpdateRolesRequestFromJSONTyped(json, false);
}
exports.UpdateRolesRequestFromJSON = UpdateRolesRequestFromJSON;
function UpdateRolesRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': json['name'],
        'description': !(0, runtime_js_1.exists)(json, 'description') ? undefined : json['description'],
        'key': json['key'],
        'isDefaultRole': !(0, runtime_js_1.exists)(json, 'is_default_role') ? undefined : json['is_default_role'],
    };
}
exports.UpdateRolesRequestFromJSONTyped = UpdateRolesRequestFromJSONTyped;
function UpdateRolesRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'description': value.description,
        'key': value.key,
        'is_default_role': value.isDefaultRole,
    };
}
exports.UpdateRolesRequestToJSON = UpdateRolesRequestToJSON;
