// @kinde-oss/kinde-typescript-sdk/dist-cjs/models/UpdateOrganizationUsersRequestUsersInner.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdateOrganizationUsersRequestUsersInnerToJSON = exports.UpdateOrganizationUsersRequestUsersInnerFromJSONTyped = exports.UpdateOrganizationUsersRequestUsersInnerFromJSON = exports.instanceOfUpdateOrganizationUsersRequestUsersInner = void 0;
var runtime_js_1 = require("../runtime.js");
/**
 * Check if a given object implements the UpdateOrganizationUsersRequestUsersInner interface.
 */
function instanceOfUpdateOrganizationUsersRequestUsersInner(value) {
    var isInstance = true;
    return isInstance;
}
exports.instanceOfUpdateOrganizationUsersRequestUsersInner = instanceOfUpdateOrganizationUsersRequestUsersInner;
function UpdateOrganizationUsersRequestUsersInnerFromJSON(json) {
    return UpdateOrganizationUsersRequestUsersInnerFromJSONTyped(json, false);
}
exports.UpdateOrganizationUsersRequestUsersInnerFromJSON = UpdateOrganizationUsersRequestUsersInnerFromJSON;
function UpdateOrganizationUsersRequestUsersInnerFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'id': !(0, runtime_js_1.exists)(json, 'id') ? undefined : json['id'],
        'operation': !(0, runtime_js_1.exists)(json, 'operation') ? undefined : json['operation'],
        'roles': !(0, runtime_js_1.exists)(json, 'roles') ? undefined : json['roles'],
        'permissions': !(0, runtime_js_1.exists)(json, 'permissions') ? undefined : json['permissions'],
    };
}
exports.UpdateOrganizationUsersRequestUsersInnerFromJSONTyped = UpdateOrganizationUsersRequestUsersInnerFromJSONTyped;
function UpdateOrganizationUsersRequestUsersInnerToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'id': value.id,
        'operation': value.operation,
        'roles': value.roles,
        'permissions': value.permissions,
    };
}
exports.UpdateOrganizationUsersRequestUsersInnerToJSON = UpdateOrganizationUsersRequestUsersInnerToJSON;
