// @kinde-oss/kinde-typescript-sdk/dist-cjs/models/UpdateRolePermissionsRequestPermissionsInner.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdateRolePermissionsRequestPermissionsInnerToJSON = exports.UpdateRolePermissionsRequestPermissionsInnerFromJSONTyped = exports.UpdateRolePermissionsRequestPermissionsInnerFromJSON = exports.instanceOfUpdateRolePermissionsRequestPermissionsInner = void 0;
var runtime_js_1 = require("../runtime.js");
/**
 * Check if a given object implements the UpdateRolePermissionsRequestPermissionsInner interface.
 */
function instanceOfUpdateRolePermissionsRequestPermissionsInner(value) {
    var isInstance = true;
    return isInstance;
}
exports.instanceOfUpdateRolePermissionsRequestPermissionsInner = instanceOfUpdateRolePermissionsRequestPermissionsInner;
function UpdateRolePermissionsRequestPermissionsInnerFromJSON(json) {
    return UpdateRolePermissionsRequestPermissionsInnerFromJSONTyped(json, false);
}
exports.UpdateRolePermissionsRequestPermissionsInnerFromJSON = UpdateRolePermissionsRequestPermissionsInnerFromJSON;
function UpdateRolePermissionsRequestPermissionsInnerFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'id': !(0, runtime_js_1.exists)(json, 'id') ? undefined : json['id'],
        'operation': !(0, runtime_js_1.exists)(json, 'operation') ? undefined : json['operation'],
    };
}
exports.UpdateRolePermissionsRequestPermissionsInnerFromJSONTyped = UpdateRolePermissionsRequestPermissionsInnerFromJSONTyped;
function UpdateRolePermissionsRequestPermissionsInnerToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'id': value.id,
        'operation': value.operation,
    };
}
exports.UpdateRolePermissionsRequestPermissionsInnerToJSON = UpdateRolePermissionsRequestPermissionsInnerToJSON;
