// @kinde-oss/kinde-typescript-sdk/dist-cjs/models/UpdateUserRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdateUserRequestToJSON = exports.UpdateUserRequestFromJSONTyped = exports.UpdateUserRequestFromJSON = exports.instanceOfUpdateUserRequest = void 0;
var runtime_js_1 = require("../runtime.js");
/**
 * Check if a given object implements the UpdateUserRequest interface.
 */
function instanceOfUpdateUserRequest(value) {
    var isInstance = true;
    return isInstance;
}
exports.instanceOfUpdateUserRequest = instanceOfUpdateUserRequest;
function UpdateUserRequestFromJSON(json) {
    return UpdateUserRequestFromJSONTyped(json, false);
}
exports.UpdateUserRequestFromJSON = UpdateUserRequestFromJSON;
function UpdateUserRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'givenName': !(0, runtime_js_1.exists)(json, 'given_name') ? undefined : json['given_name'],
        'familyName': !(0, runtime_js_1.exists)(json, 'family_name') ? undefined : json['family_name'],
        'isSuspended': !(0, runtime_js_1.exists)(json, 'is_suspended') ? undefined : json['is_suspended'],
        'isPasswordResetRequested': !(0, runtime_js_1.exists)(json, 'is_password_reset_requested') ? undefined : json['is_password_reset_requested'],
    };
}
exports.UpdateUserRequestFromJSONTyped = UpdateUserRequestFromJSONTyped;
function UpdateUserRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'given_name': value.givenName,
        'family_name': value.familyName,
        'is_suspended': value.isSuspended,
        'is_password_reset_requested': value.isPasswordResetRequested,
    };
}
exports.UpdateUserRequestToJSON = UpdateUserRequestToJSON;
