// @kinde-oss/kinde-typescript-sdk/dist-cjs/models/UserIdentityResult.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserIdentityResultToJSON = exports.UserIdentityResultFromJSONTyped = exports.UserIdentityResultFromJSON = exports.instanceOfUserIdentityResult = void 0;
var runtime_js_1 = require("../runtime.js");
/**
 * Check if a given object implements the UserIdentityResult interface.
 */
function instanceOfUserIdentityResult(value) {
    var isInstance = true;
    return isInstance;
}
exports.instanceOfUserIdentityResult = instanceOfUserIdentityResult;
function UserIdentityResultFromJSON(json) {
    return UserIdentityResultFromJSONTyped(json, false);
}
exports.UserIdentityResultFromJSON = UserIdentityResultFromJSON;
function UserIdentityResultFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'created': !(0, runtime_js_1.exists)(json, 'created') ? undefined : json['created'],
    };
}
exports.UserIdentityResultFromJSONTyped = UserIdentityResultFromJSONTyped;
function UserIdentityResultToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'created': value.created,
    };
}
exports.UserIdentityResultToJSON = UserIdentityResultToJSON;
