// @kinde-oss/kinde-typescript-sdk/dist-cjs/models/UserProfile.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserProfileToJSON = exports.UserProfileFromJSONTyped = exports.UserProfileFromJSON = exports.instanceOfUserProfile = void 0;
var runtime_js_1 = require("../runtime.js");
/**
 * Check if a given object implements the UserProfile interface.
 */
function instanceOfUserProfile(value) {
    var isInstance = true;
    return isInstance;
}
exports.instanceOfUserProfile = instanceOfUserProfile;
function UserProfileFromJSON(json) {
    return UserProfileFromJSONTyped(json, false);
}
exports.UserProfileFromJSON = UserProfileFromJSON;
function UserProfileFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'id': !(0, runtime_js_1.exists)(json, 'id') ? undefined : json['id'],
        'preferredEmail': !(0, runtime_js_1.exists)(json, 'preferred_email') ? undefined : json['preferred_email'],
        'providedId': !(0, runtime_js_1.exists)(json, 'provided_id') ? undefined : json['provided_id'],
        'lastName': !(0, runtime_js_1.exists)(json, 'last_name') ? undefined : json['last_name'],
        'firstName': !(0, runtime_js_1.exists)(json, 'first_name') ? undefined : json['first_name'],
        'picture': !(0, runtime_js_1.exists)(json, 'picture') ? undefined : json['picture'],
    };
}
exports.UserProfileFromJSONTyped = UserProfileFromJSONTyped;
function UserProfileToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'id': value.id,
        'preferred_email': value.preferredEmail,
        'provided_id': value.providedId,
        'last_name': value.lastName,
        'first_name': value.firstName,
        'picture': value.picture,
    };
}
exports.UserProfileToJSON = UserProfileToJSON;
