// @kinde-oss/kinde-typescript-sdk/dist-cjs/models/UsersResponse.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UsersResponseToJSON = exports.UsersResponseFromJSONTyped = exports.UsersResponseFromJSON = exports.instanceOfUsersResponse = void 0;
var runtime_js_1 = require("../runtime.js");
var UsersResponseUsersInner_js_1 = require("./UsersResponseUsersInner.js");
/**
 * Check if a given object implements the UsersResponse interface.
 */
function instanceOfUsersResponse(value) {
    var isInstance = true;
    return isInstance;
}
exports.instanceOfUsersResponse = instanceOfUsersResponse;
function UsersResponseFromJSON(json) {
    return UsersResponseFromJSONTyped(json, false);
}
exports.UsersResponseFromJSON = UsersResponseFromJSON;
function UsersResponseFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'code': !(0, runtime_js_1.exists)(json, 'code') ? undefined : json['code'],
        'message': !(0, runtime_js_1.exists)(json, 'message') ? undefined : json['message'],
        'users': !(0, runtime_js_1.exists)(json, 'users') ? undefined : (json['users'].map(UsersResponseUsersInner_js_1.UsersResponseUsersInnerFromJSON)),
        'nextToken': !(0, runtime_js_1.exists)(json, 'next_token') ? undefined : json['next_token'],
    };
}
exports.UsersResponseFromJSONTyped = UsersResponseFromJSONTyped;
function UsersResponseToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'code': value.code,
        'message': value.message,
        'users': value.users === undefined ? undefined : (value.users.map(UsersResponseUsersInner_js_1.UsersResponseUsersInnerToJSON)),
        'next_token': value.nextToken,
    };
}
exports.UsersResponseToJSON = UsersResponseToJSON;
