// @kinde-oss/kinde-typescript-sdk/dist-cjs/models/UsersResponseUsersInner.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UsersResponseUsersInnerToJSON = exports.UsersResponseUsersInnerFromJSONTyped = exports.UsersResponseUsersInnerFromJSON = exports.instanceOfUsersResponseUsersInner = void 0;
var runtime_js_1 = require("../runtime.js");
var UserIdentitiesInner_js_1 = require("./UserIdentitiesInner.js");
/**
 * Check if a given object implements the UsersResponseUsersInner interface.
 */
function instanceOfUsersResponseUsersInner(value) {
    var isInstance = true;
    return isInstance;
}
exports.instanceOfUsersResponseUsersInner = instanceOfUsersResponseUsersInner;
function UsersResponseUsersInnerFromJSON(json) {
    return UsersResponseUsersInnerFromJSONTyped(json, false);
}
exports.UsersResponseUsersInnerFromJSON = UsersResponseUsersInnerFromJSON;
function UsersResponseUsersInnerFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'id': !(0, runtime_js_1.exists)(json, 'id') ? undefined : json['id'],
        'providedId': !(0, runtime_js_1.exists)(json, 'provided_id') ? undefined : json['provided_id'],
        'email': !(0, runtime_js_1.exists)(json, 'email') ? undefined : json['email'],
        'lastName': !(0, runtime_js_1.exists)(json, 'last_name') ? undefined : json['last_name'],
        'firstName': !(0, runtime_js_1.exists)(json, 'first_name') ? undefined : json['first_name'],
        'isSuspended': !(0, runtime_js_1.exists)(json, 'is_suspended') ? undefined : json['is_suspended'],
        'picture': !(0, runtime_js_1.exists)(json, 'picture') ? undefined : json['picture'],
        'totalSignIns': !(0, runtime_js_1.exists)(json, 'total_sign_ins') ? undefined : json['total_sign_ins'],
        'failedSignIns': !(0, runtime_js_1.exists)(json, 'failed_sign_ins') ? undefined : json['failed_sign_ins'],
        'lastSignedIn': !(0, runtime_js_1.exists)(json, 'last_signed_in') ? undefined : json['last_signed_in'],
        'createdOn': !(0, runtime_js_1.exists)(json, 'created_on') ? undefined : json['created_on'],
        'organizations': !(0, runtime_js_1.exists)(json, 'organizations') ? undefined : json['organizations'],
        'identities': !(0, runtime_js_1.exists)(json, 'identities') ? undefined : (json['identities'].map(UserIdentitiesInner_js_1.UserIdentitiesInnerFromJSON)),
    };
}
exports.UsersResponseUsersInnerFromJSONTyped = UsersResponseUsersInnerFromJSONTyped;
function UsersResponseUsersInnerToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'id': value.id,
        'provided_id': value.providedId,
        'email': value.email,
        'last_name': value.lastName,
        'first_name': value.firstName,
        'is_suspended': value.isSuspended,
        'picture': value.picture,
        'total_sign_ins': value.totalSignIns,
        'failed_sign_ins': value.failedSignIns,
        'last_signed_in': value.lastSignedIn,
        'created_on': value.createdOn,
        'organizations': value.organizations,
        'identities': value.identities === undefined ? undefined : (value.identities.map(UserIdentitiesInner_js_1.UserIdentitiesInnerToJSON)),
    };
}
exports.UsersResponseUsersInnerToJSON = UsersResponseUsersInnerToJSON;
