// @kinde-oss/kinde-typescript-sdk/dist-cjs/models/UserProfileV2.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserProfileV2ToJSON = exports.UserProfileV2FromJSONTyped = exports.UserProfileV2FromJSON = exports.instanceOfUserProfileV2 = void 0;
var runtime_js_1 = require("../runtime.js");
/**
 * Check if a given object implements the UserProfileV2 interface.
 */
function instanceOfUserProfileV2(value) {
    var isInstance = true;
    return isInstance;
}
exports.instanceOfUserProfileV2 = instanceOfUserProfileV2;
function UserProfileV2FromJSON(json) {
    return UserProfileV2FromJSONTyped(json, false);
}
exports.UserProfileV2FromJSON = UserProfileV2FromJSON;
function UserProfileV2FromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'id': !(0, runtime_js_1.exists)(json, 'id') ? undefined : json['id'],
        'sub': !(0, runtime_js_1.exists)(json, 'sub') ? undefined : json['sub'],
        'providedId': !(0, runtime_js_1.exists)(json, 'provided_id') ? undefined : json['provided_id'],
        'name': !(0, runtime_js_1.exists)(json, 'name') ? undefined : json['name'],
        'givenName': !(0, runtime_js_1.exists)(json, 'given_name') ? undefined : json['given_name'],
        'familyName': !(0, runtime_js_1.exists)(json, 'family_name') ? undefined : json['family_name'],
        'updatedAt': !(0, runtime_js_1.exists)(json, 'updated_at') ? undefined : json['updated_at'],
        'email': !(0, runtime_js_1.exists)(json, 'email') ? undefined : json['email'],
        'picture': !(0, runtime_js_1.exists)(json, 'picture') ? undefined : json['picture'],
    };
}
exports.UserProfileV2FromJSONTyped = UserProfileV2FromJSONTyped;
function UserProfileV2ToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'id': value.id,
        'sub': value.sub,
        'provided_id': value.providedId,
        'name': value.name,
        'given_name': value.givenName,
        'family_name': value.familyName,
        'updated_at': value.updatedAt,
        'email': value.email,
        'picture': value.picture,
    };
}
exports.UserProfileV2ToJSON = UserProfileV2ToJSON;
