// @kinde-oss/kinde-typescript-sdk/dist-cjs/models/UserIdentity.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserIdentityToJSON = exports.UserIdentityFromJSONTyped = exports.UserIdentityFromJSON = exports.instanceOfUserIdentity = void 0;
var runtime_js_1 = require("../runtime.js");
var UserIdentityResult_js_1 = require("./UserIdentityResult.js");
/**
 * Check if a given object implements the UserIdentity interface.
 */
function instanceOfUserIdentity(value) {
    var isInstance = true;
    return isInstance;
}
exports.instanceOfUserIdentity = instanceOfUserIdentity;
function UserIdentityFromJSON(json) {
    return UserIdentityFromJSONTyped(json, false);
}
exports.UserIdentityFromJSON = UserIdentityFromJSON;
function UserIdentityFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'type': !(0, runtime_js_1.exists)(json, 'type') ? undefined : json['type'],
        'result': !(0, runtime_js_1.exists)(json, 'result') ? undefined : (0, UserIdentityResult_js_1.UserIdentityResultFromJSON)(json['result']),
    };
}
exports.UserIdentityFromJSONTyped = UserIdentityFromJSONTyped;
function UserIdentityToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'type': value.type,
        'result': (0, UserIdentityResult_js_1.UserIdentityResultToJSON)(value.result),
    };
}
exports.UserIdentityToJSON = UserIdentityToJSON;
