// @kinde-oss/kinde-typescript-sdk/dist-cjs/models/UserIdentitiesInner.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserIdentitiesInnerToJSON = exports.UserIdentitiesInnerFromJSONTyped = exports.UserIdentitiesInnerFromJSON = exports.instanceOfUserIdentitiesInner = void 0;
var runtime_js_1 = require("../runtime.js");
/**
 * Check if a given object implements the UserIdentitiesInner interface.
 */
function instanceOfUserIdentitiesInner(value) {
    var isInstance = true;
    return isInstance;
}
exports.instanceOfUserIdentitiesInner = instanceOfUserIdentitiesInner;
function UserIdentitiesInnerFromJSON(json) {
    return UserIdentitiesInnerFromJSONTyped(json, false);
}
exports.UserIdentitiesInnerFromJSON = UserIdentitiesInnerFromJSON;
function UserIdentitiesInnerFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'type': !(0, runtime_js_1.exists)(json, 'type') ? undefined : json['type'],
        'identity': !(0, runtime_js_1.exists)(json, 'identity') ? undefined : json['identity'],
    };
}
exports.UserIdentitiesInnerFromJSONTyped = UserIdentitiesInnerFromJSONTyped;
function UserIdentitiesInnerToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'type': value.type,
        'identity': value.identity,
    };
}
exports.UserIdentitiesInnerToJSON = UserIdentitiesInnerToJSON;
