// @kinde-oss/kinde-typescript-sdk/dist/sdk/utilities/token-utils.js
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __generator = (this && this.__generator) || function (thisArg, body) {
    var _ = { label: 0, sent: function() { if (t[0] & 1) throw t[1]; return t[1]; }, trys: [], ops: [] }, f, y, t, g;
    return g = { next: verb(0), "throw": verb(1), "return": verb(2) }, typeof Symbol === "function" && (g[Symbol.iterator] = function() { return this; }), g;
    function verb(n) { return function (v) { return step([n, v]); }; }
    function step(op) {
        if (f) throw new TypeError("Generator is already executing.");
        while (g && (g = 0, op[0] && (_ = 0)), _) try {
            if (f = 1, y && (t = op[0] & 2 ? y["return"] : op[0] ? y["throw"] || ((t = y["return"]) && t.call(y), 0) : y.next) && !(t = t.call(y, op[1])).done) return t;
            if (y = 0, t) op = [op[0] & 2, t.value];
            switch (op[0]) {
                case 0: case 1: t = op; break;
                case 4: _.label++; return { value: op[1], done: false };
                case 5: _.label++; y = op[1]; op = [0]; continue;
                case 7: op = _.ops.pop(); _.trys.pop(); continue;
                default:
                    if (!(t = _.trys, t = t.length > 0 && t[t.length - 1]) && (op[0] === 6 || op[0] === 2)) { _ = 0; continue; }
                    if (op[0] === 3 && (!t || (op[1] > t[0] && op[1] < t[3]))) { _.label = op[1]; break; }
                    if (op[0] === 6 && _.label < t[1]) { _.label = t[1]; t = op; break; }
                    if (t && _.label < t[2]) { _.label = t[2]; _.ops.push(op); break; }
                    if (t[2]) _.ops.pop();
                    _.trys.pop(); continue;
            }
            op = body.call(thisArg, _);
        } catch (e) { op = [6, e]; y = 0; } finally { f = t = 0; }
        if (op[0] & 5) throw op[1]; return { value: op[0] ? op[1] : void 0, done: true };
    }
};
import { jwtDecode } from 'jwt-decode';
/**
 * Extracts the payload from the provided idToken and saves the extracted
 * payload to the current session.
 * @param {SessionManager} sessionManager
 * @param {string} idToken
 * @returns {void}
 */
var commitUserToMemoryFromToken = function (sessionManager, idToken) { return __awaiter(void 0, void 0, void 0, function () {
    var idTokenPayload, user;
    var _a;
    return __generator(this, function (_b) {
        switch (_b.label) {
            case 0:
                idTokenPayload = jwtDecode(idToken);
                user = {
                    family_name: idTokenPayload.family_name,
                    given_name: idTokenPayload.given_name,
                    picture: (_a = idTokenPayload.picture) !== null && _a !== void 0 ? _a : null,
                    email: idTokenPayload.email,
                    id: idTokenPayload.sub,
                };
                return [4 /*yield*/, sessionManager.setSessionItem('user', user)];
            case 1:
                _b.sent();
                return [2 /*return*/];
        }
    });
}); };
/**
 * Saves the provided token and its extracted payload to the current session.
 * @param {SessionManager} sessionManager
 * @param {string} token
 * @param {TokenType} type
 */
export var commitTokenToMemory = function (sessionManager, token, type) { return __awaiter(void 0, void 0, void 0, function () {
    return __generator(this, function (_a) {
        switch (_a.label) {
            case 0: return [4 /*yield*/, sessionManager.setSessionItem(type, token)];
            case 1:
                _a.sent();
                if (!(type === 'id_token')) return [3 /*break*/, 3];
                return [4 /*yield*/, commitUserToMemoryFromToken(sessionManager, token)];
            case 2:
                _a.sent();
                _a.label = 3;
            case 3: return [2 /*return*/];
        }
    });
}); };
/**
 * Saves the access, refresh and id tokens provided in the `TokenCollection`
 * object to the current session.
 * @param {SessionManager} sessionManager
 * @param tokens
 */
export var commitTokensToMemory = function (sessionManager, tokens) { return __awaiter(void 0, void 0, void 0, function () {
    return __generator(this, function (_a) {
        switch (_a.label) {
            case 0: return [4 /*yield*/, Promise.all([
                    commitTokenToMemory(sessionManager, tokens.refresh_token, 'refresh_token'),
                    commitTokenToMemory(sessionManager, tokens.access_token, 'access_token'),
                    commitTokenToMemory(sessionManager, tokens.id_token, 'id_token'),
                ])];
            case 1:
                _a.sent();
                return [2 /*return*/];
        }
    });
}); };
/**
 * Extracts the refresh token from current session returns null if the
 * token is not found.
 * @param {SessionManager} sessionManager
 * @returns {string | null}
 */
export var getRefreshToken = function (sessionManager) { return __awaiter(void 0, void 0, void 0, function () {
    return __generator(this, function (_a) {
        switch (_a.label) {
            case 0: return [4 /*yield*/, sessionManager.getSessionItem('refresh_token')];
            case 1: return [2 /*return*/, _a.sent()];
        }
    });
}); };
/**
 * Extracts the access token from current session returns null if the
 * token is not found.
 * @param {SessionManager} sessionManager
 * @returns {string | null}
 */
export var getAccessToken = function (sessionManager) { return __awaiter(void 0, void 0, void 0, function () {
    return __generator(this, function (_a) {
        switch (_a.label) {
            case 0: return [4 /*yield*/, sessionManager.getSessionItem('access_token')];
            case 1: return [2 /*return*/, _a.sent()];
        }
    });
}); };
/**
 * Extracts the user information from the current session returns null if
 * the token is not found.
 * @param {SessionManager} sessionManager
 * @returns {string | null}
 */
export var getUserFromMemory = function (sessionManager) { return __awaiter(void 0, void 0, void 0, function () {
    return __generator(this, function (_a) {
        switch (_a.label) {
            case 0: return [4 /*yield*/, sessionManager.getSessionItem('user')];
            case 1: return [2 /*return*/, _a.sent()];
        }
    });
}); };
/**
 * Saves the provided user details as `UserType` to the current session.
 * @param {SessionManager} sessionManager
 * @param {UserType} user
 */
export var commitUserToMemory = function (sessionManager, user) { return __awaiter(void 0, void 0, void 0, function () {
    return __generator(this, function (_a) {
        switch (_a.label) {
            case 0: return [4 /*yield*/, sessionManager.setSessionItem('user', user)];
            case 1:
                _a.sent();
                return [2 /*return*/];
        }
    });
}); };
/**
 * Checks if the provided JWT token is valid (expired or not).
 * @param {string | null} token
 * @returns {boolean} is expired or not
 */
export var isTokenExpired = function (token) {
    if (!token)
        return true;
    var currentUnixTime = Math.floor(Date.now() / 1000);
    var tokenPayload = jwtDecode(token);
    return currentUnixTime >= tokenPayload.exp;
};
